<h1><center>Obtaining Stocks and Market Data + Fama-MacBeth Regression</center></h1>

## Course: Introduction to Programming 2477-2223 S1, Nova SBE


### Team:
*  ### Daniel Brosko (55036)
*  ### Joao Victor Rosa Omori (53135)
*  ### Marion Chevalier Montebrun (55246)
*  ### Parsa Rahimi Fetrati (54283)

## 1 Introduction

Portfolio analysis is a commonly used tool for asset pricing. It is useful for illustrating detailed information on the cross-sectional relationship between variables, and it also helps to uncover the relationship between those variables and future stock returns (Bali, Engle, and Murray, 2016). The biggest advantage for portfolio analysis is that it does not require any assumptions as it is a nonparametric approach. Because of its nonparametric characteristics, it could detect nonlinear relationship between variables as well (Bali, Engle, and Murray, 2016).

This paper will focus on Fama-MacBeth regression. The Fama-MacBeth regression analysis is to investigate the features of variable coefficients in the cross-sectional regression on return and targeted variables. Bai and Zhou (2015) applied the Fama-MacBeth regression on samples with different sizes and time ranges. They discovered the significant impact of sample size on stock returns and risk premium estimates.

The exact variables we examine the effect of are market return, short time reversal, and size of the firm (logarithm of market capital). We obtained the Market Cap data by scraping the macrotrends.net website, stocks data from yahoo finance (we found working source despite the official one being degraded), and the market data from professor Kenneth French's website.

For the analysis, we use 20 stocks from S&P 500 in this project. Sections 3 is focused on obtaining the data. Intermediate computations are covered in Section 4. In Section 5, we finally perform the Fama-MacBeth regression, and comment on findings of the model results.

The period on which we were doing our analyses covers the period from January 1, 2010 till October 31, 2022. We started working with daily prices and returns for beta estimates. Then we switched to the monthly closing data for further analyses covered here. The market data (F-F factors and Market Cap), were all downloaded in daily format and afterwards transformed into monthly data. Additionals comments are present across the particular "chunks" of the code, and the particular variables are properly defined in the "Empirical Methodology".

### Empirical methodology

The parameters in FM regression are estimated in two steps:

* First regress each of n asset returns against m proposed risk factors to determine each asset's beta exposures.

\begin{array}{lcr}R_{1,t}=\alpha _{1}+\beta _{1,F_{1}}F_{1,t}+\beta _{1,F_{2}}F_{2,t}+\cdots +\beta _{1,F_{m}}F_{m,t}+\epsilon _{1,t}\\R_{2,t}=\alpha _{2}+\beta _{2,F_{1}}F_{1,t}+\beta _{2,F_{2}}F_{2,t}+\cdots +\beta _{2,F_{m}}F_{m,t}+\epsilon _{2,t}\\\vdots \\R_{n,t}=\alpha _{n}+\beta _{n,F_{1}}F_{1,t}+\beta _{n,F_{2}}F_{2,t}+\cdots +\beta _{n,F_{m}}F_{m,t}+\epsilon _{n,t}\end{array}

* Then regress all asset returns for each of T time periods against the previously estimated betas to determine the risk premium for each factor.

\begin{array}{lcr}R_{i,1}=\gamma _{1,0}+\gamma _{1,1}{\hat {\beta }}_{i,F_{1}}+\gamma _{1,2}{\hat {\beta }}_{i,F_{2}}+\cdots +\gamma _{1,m}{\hat {\beta }}_{i,F_{m}}+\epsilon _{i,1}\\R_{i,2}=\gamma _{2,0}+\gamma _{2,1}{\hat {\beta }}_{i,F_{1}}+\gamma _{2,2}{\hat {\beta }}_{i,F_{2}}+\cdots +\gamma _{2,m}{\hat {\beta }}_{i,F_{m}}+\epsilon _{i,2}\\\vdots \\R_{i,T}=\gamma _{T,0}+\gamma _{T,1}{\hat {\beta }}_{i,F_{1}}+\gamma _{T,2}{\hat {\beta }}_{i,F_{2}}+\cdots +\gamma _{T,m}{\hat {\beta }}_{i,F_{m}}+\epsilon _{i,T}\end{array} (source: https://en.wikipedia.org/wiki/Fama–MacBeth_regression.)


From coding point of view, the main part is this notebook, while we also use .py scripts written from the scratch.
The project is also available at GitHub: https://github.com/tigerbroko/Nova-SBE-Introduction-to-Programming.git (SSH: git@github.com:tigerbroko/Nova-SBE-Introduction-to-Programming.git).

## 2 General packages loading

In [1]:
import pandas as pd
import numpy as np
import re
import requests
from bs4 import BeautifulSoup
import datetime
import time
import yfinance as yf
from sklearn.linear_model import LinearRegression
import statistics

#importing own functions
import scraping_functions as sf
import computing_functions as cf
import stock_prices as sp

In [2]:
# pip list # used for requirements.txt to get the version numbers

In [3]:
# loading tickers of selected stocks
tickers = sf.load_tickers("tickers_set.csv")

## 3 Obtaining the Data
### 3.1 Market Capitalization (Size) Data Scraping

Here we scrape the market capitalizations of the firms using own scraping function from scraping_function.py.
The data are in billions of USD.

In [4]:
starting_time = time.time()

# we perform the scraping of market capitalization for all desired stocks 
mkt_cap_df_list = []

for ticker in tickers:
    mkt_cap_df_list.append(sf.mkt_cap_scraper(ticker))


elapsed_time = time.time() - starting_time
print('Elapsed time:', round(elapsed_time,3) , 'seconds.')

# mkt_cap_df_list

Elapsed time: 1.498 seconds.


In [ ]:
# merging mkt. capital for all stocks, and renaming the columns according to particular stocks
df_mkt_cap = mkt_cap_df_list[0]

for dataframe in mkt_cap_df_list[1:] :
    df_mkt_cap = df_mkt_cap.merge(dataframe, on='Date', how = 'inner')

mc_columns_names = ['Date']
mc_columns_names.extend(tickers)
df_mkt_cap.columns = mc_columns_names
df_mkt_cap.set_index('Date', inplace=True)

# using size instead of mkt_cap to rule out skewness
df_size = np.log(df_mkt_cap + 0.000001)

### 3.2 Obtaining Stock Prices

In [6]:
# downloading stocks prices - designed to save as csv, so we have "back-up" if something goes wrong later
success_tickers, failed_tickers = sp.get_stocks(tickers)

TER - SUCCESS
ADM - SUCCESS
TMUS - SUCCESS
IDXX - SUCCESS
LNC - SUCCESS
WST - SUCCESS
AMZN - SUCCESS
DOV - SUCCESS
ORLY - SUCCESS
BSX - SUCCESS
USB - SUCCESS
ETR - SUCCESS
AZO - SUCCESS
PVH - SUCCESS
BIIB - SUCCESS
ETN - SUCCESS
PFE - SUCCESS
WMB - SUCCESS
FITB - SUCCESS
CHRW - SUCCESS
Elapsed time: 21.827 seconds.


In [7]:
# loading stock prices into list of dataframes using own function
stocks_df_list = sp.load_stocks(success_tickers)
# stocks_df_list

In [ ]:
df_close_prices = stocks_df_list[0][['Date','Adj Close']]
 
for index in range(1, len(stocks_df_list)):
    dataframe = stocks_df_list[index][['Date','Adj Close']]
    df_close_prices = df_close_prices.merge(dataframe, on='Date', how = 'inner')

cp_columns_names = ['Date']
cp_columns_names.extend(success_tickers)
df_close_prices.columns = cp_columns_names
df_close_prices['Date'] = pd.to_datetime(df_close_prices['Date']).dt.date
df_close_prices.set_index('Date', inplace=True)

In [9]:
# Making returns from daily prices
# daily_returns = df_close_prices.pct_change(1) # 1 for ONE DAY lookback
daily_returns = np.log(df_close_prices/df_close_prices.shift())
daily_returns.dropna(inplace=True)
daily_returns.head()

# monthly_return = prices.pct_change(21) # 21 for ONE MONTH lookback
# annual_return = prices.pct_change(252)

TER       ADM      TMUS      IDXX       LNC       WST  \
Date                                                                     
2009-12-09  0.011288 -0.010768 -0.026550 -0.005999  0.007538 -0.002321   
2009-12-10 -0.015425  0.000984  0.030908  0.009658  0.005286  0.006947   
2009-12-11 -0.003113 -0.000656  0.007220  0.002688 -0.014160  0.013245   
2009-12-14  0.069270 -0.008564 -0.011578  0.006878  0.012841  0.009820   
2009-12-15  0.002905  0.013472  0.067536  0.008343 -0.027204  0.001503   

                AMZN       DOV      ORLY       BSX       USB       ETR  \
Date                                                                     
2009-12-09 -0.021099 -0.005363 -0.008706  0.000000 -0.003487 -0.000494   
2009-12-10  0.030525  0.001954 -0.003451  0.027909  0.005661  0.010699   
2009-12-11 -0.009127  0.001950 -0.001064 -0.005750  0.000868  0.015897   
2009-12-14 -0.020865  0.012822  0.009273  0.029548 -0.000434  0.008392   
2009-12-15 -0.008792 -0.005786 -0.010604 -0.007870 -0.028617 -0.009115   

                 AZO       PVH      BIIB       ETN       PFE       WMB  \
Date                                                                     
2009-12-09 -0.003767 -0.018150  0.002286  0.000000  0.026668  0.000517   
2009-12-10 -0.005839  0.007098  0.001867  0.005139  0.004377  0.013864   
2009-12-11  0.002442  0.022231  0.008049  0.001242 -0.001093 -0.001020   
2009-12-14  0.017688  0.009099  0.004512  0.019355  0.005449  0.036586   
2009-12-15 -0.010461 -0.012068  0.011394 -0.001523 -0.005996  0.013200   

                FITB      CHRW  
Date                            
2009-12-09  0.009766  0.000871  
2009-12-10 -0.020619  0.003995  
2009-12-11  0.009872  0.007082  
2009-12-14  0.014627  0.016050  
2009-12-15 -0.052670  0.011284

### 3.3 Market Return from prof. Kenneth French's website 

We dowloaded data about the market from https://mba.tuck.dartmouth.edu/pages/faculty/ken.french/data_library.html , however, we need to change the date into suitable format.

In [10]:
data_market = pd.read_csv("F-F_Research_Data_Factors_daily.CSV", skiprows=3)
data_market = data_market[:-1]

data_market.rename(columns = {'Unnamed: 0': 'Date'}, inplace = True)

# using simple lambda function to put dates into correct format
date_format_lambda = lambda x: datetime.datetime.strptime(x, '%Y%m%d')
data_market['Date'] = data_market['Date'].map(date_format_lambda)
data_market['Date'] = pd.to_datetime(data_market['Date']).dt.date
data_market.set_index('Date', inplace=True)
data_market.head()

Mkt-RF   SMB   HML     RF
Date                                 
1926-07-01    0.10 -0.25 -0.27  0.009
1926-07-02    0.45 -0.33 -0.06  0.009
1926-07-06    0.17  0.30 -0.39  0.009
1926-07-07    0.09 -0.58  0.02  0.009
1926-07-08    0.21 -0.38  0.19  0.009

### 3.4 Short-Time-Reversal from prof. Kenneth French's website 
We dowloaded STR data from https://mba.tuck.dartmouth.edu/pages/faculty/ken.french/data_library.html
and proceed in the similar fashion to subsection 3.3.

In [11]:
st_reversal = pd.read_csv("F-F_ST_Reversal_Factor_daily.CSV", skiprows=12)
st_reversal = st_reversal[:-1]
st_reversal.rename(columns = {'Unnamed: 0': 'Date'}, inplace = True)

# using simple lambda function to put dates into correct format
date_format_lambda = lambda x: datetime.datetime.strptime(x, '%Y%m%d')
st_reversal['Date'] = st_reversal['Date'].map(date_format_lambda)
st_reversal['Date'] = pd.to_datetime(st_reversal['Date']).dt.date
st_reversal.set_index('Date', inplace=True)
st_reversal.head()

ST_Rev
Date              
1926-01-26    0.14
1926-01-27    0.62
1926-01-28    0.22
1926-01-29    0.58
1926-01-30    0.46

### 3.4 Filtering data for reasonable time interval

In [12]:
size = df_size.query("Date >= datetime.date(2010,1,1) and Date <= datetime.date(2022,10,31)")
returns = daily_returns.query("Date >= datetime.date(2010,1,1) and Date <= datetime.date(2022,10,31)")
market = data_market.query("Date >= datetime.date(2010,1,1) and Date <= datetime.date(2022,10,31)")
reversal = st_reversal.query("Date >= datetime.date(2010,1,1) and Date <= datetime.date(2022,10,31)")

Here, we adjust stock returns for risk-free rate, to simplify the linear regression (we get rid of the intercept, which is rf rate).

In [13]:
# adjusting stock returns to risk-free rate
risk_free_rate = market.loc[:, 'RF']
risk_free_rate = pd.DataFrame(risk_free_rate)
risk_free_rate_column = risk_free_rate.copy()

for column in range(1, len(success_tickers)):
    risk_free_rate = pd.concat([risk_free_rate, risk_free_rate_column], axis=1)

excess_returns = returns.subtract(risk_free_rate.values)

excess_returns

TER       ADM      TMUS      IDXX       LNC       WST  \
Date                                                                     
2010-01-04  0.021209  0.005097  0.043597  0.011718  0.039406  0.006864   
2010-01-05  0.003643  0.005387  0.002506  0.001663  0.020651 -0.012749   
2010-01-06 -0.009132 -0.002532 -0.069977  0.002397  0.015026 -0.009799   
2010-01-07  0.005489 -0.010511  0.009352  0.037235  0.023591  0.003105   
2010-01-08  0.010890 -0.012567 -0.057471 -0.008016 -0.015419 -0.000258   
...              ...       ...       ...       ...       ...       ...   
2022-10-25  0.016880  0.009619 -0.001641  0.010255  0.013569  0.040840   
2022-10-26  0.021121  0.015527 -0.009508  0.005746 -0.027826  0.009612   
2022-10-27 -0.040488 -0.005247 -0.012492 -0.017047 -0.000641 -0.150705   
2022-10-28  0.031773 -0.003064  0.060147  0.011599  0.019633  0.008162   
2022-10-31 -0.035288  0.010892 -0.007298 -0.023708 -0.013966  0.009150   

                AMZN       DOV      ORLY       BSX       USB       ETR  \
Date                                                                     
2010-01-04 -0.004620  0.017864  0.009919  0.001111  0.016304  0.003781   
2010-01-05  0.005883 -0.004733  0.001557  0.003324  0.025033 -0.016075   
2010-01-06 -0.018282  0.011790 -0.004419  0.013187  0.010600  0.005059   
2010-01-07 -0.017160  0.021796 -0.003131 -0.007671  0.019215 -0.012010   
2010-01-08  0.026717  0.028266 -0.008924 -0.009950  0.001654 -0.009638   
...              ...       ...       ...       ...       ...       ...   
2022-10-25 -0.004511 -0.002004 -0.001584 -0.003222  0.010170  0.012276   
2022-10-26 -0.052824 -0.008881 -0.008772  0.000795 -0.001995 -0.011286   
2022-10-27 -0.052485 -0.007322  0.033969 -0.016038 -0.018340 -0.008044   
2022-10-28 -0.081468  0.018704  0.016986  0.026295  0.007366  0.014382   
2022-10-31 -0.020424 -0.020140 -0.009960 -0.012159 -0.020612 -0.016492   

                 AZO       PVH      BIIB       ETN       PFE       WMB  \
Date                                                                     
2010-01-04 -0.000253  0.010272  0.002613  0.010943  0.039876  0.024832   
2010-01-05 -0.008388  0.036784 -0.004859 -0.005925 -0.014366  0.012417   
2010-01-06 -0.009425  0.023640  0.000936 -0.004389 -0.003221  0.031934   
2010-01-07  0.013182  0.000000 -0.008269  0.014658 -0.003770 -0.007108   
2010-01-08 -0.012925 -0.001605  0.020361  0.022198  0.008062  0.007993   
...              ...       ...       ...       ...       ...       ...   
2022-10-25  0.002356  0.047404 -0.014867  0.020155 -0.009903  0.012830   
2022-10-26 -0.017666 -0.004292  0.013803 -0.014635 -0.000743 -0.001625   
2022-10-27  0.028674 -0.027370 -0.020171 -0.002992 -0.017972 -0.000172   
2022-10-28 -0.002420  0.004988  0.011842  0.013179  0.025282 -0.005783   
2022-10-31 -0.015184 -0.030489 -0.013994 -0.012398 -0.029728 -0.009165   

                FITB      CHRW  
Date                            
2010-01-04  0.037246  0.010333  
2010-01-05  0.026331 -0.020946  
2010-01-06  0.005758 -0.013339  
2010-01-07  0.058539 -0.009638  
2010-01-08  0.000902  0.013294  
...              ...       ...  
2022-10-25  0.017114 -0.007195  
2022-10-26 -0.006484  0.006774  
2022-10-27 -0.015516 -0.007896  
2022-10-28  0.006665 -0.000722  
2022-10-31 -0.019093 -0.011818  

[3230 rows x 20 columns]

Just to remind what objects (dataframes) are important fot us now, we obtained the following dataframes that we are going to use:

* size - logarithm of market capitalization    
* excess_returns - daily stock returns adjusted to risk-free rate
* market - FF market factors
* reversal - Short-Time-Reversal market factor

## 4 Intermediate Computations
### 4.1 Estimating Market Betas

We estimate stocks's sensitivity to the market (beta factors) on daily data using linear regression (implemented in the market_betas function from computation_functions.py script).

In [14]:
mkt_betas = cf.market_betas(excess_returns, market)
mkt_betas.head()

Elapsed time: 11.31 seconds.


TER       ADM      TMUS      IDXX       LNC       WST  \
Date                                                                     
2011-01-04  0.017935  0.007381  0.011974  0.009055  0.018687  0.009052   
2011-01-05  0.017977  0.007416  0.011807  0.009080  0.018637  0.009105   
2011-01-06  0.017931  0.007414  0.011838  0.009078  0.018613  0.009118   
2011-01-07  0.017933  0.007398  0.011895  0.009077  0.018613  0.009125   
2011-01-10  0.017924  0.007408  0.011905  0.009041  0.018591  0.009127   

                AMZN       DOV      ORLY       BSX       USB       ETR  \
Date                                                                     
2011-01-04  0.010824  0.013625  0.007077  0.009581  0.011858  0.006959   
2011-01-05  0.010932  0.013653  0.007119  0.009666  0.011879  0.006986   
2011-01-06  0.010941  0.013630  0.007117  0.009667  0.011852  0.006981   
2011-01-07  0.010952  0.013630  0.007133  0.009669  0.011856  0.006982   
2011-01-10  0.010977  0.013611  0.007139  0.009693  0.011842  0.006996   

                 AZO       PVH      BIIB       ETN       PFE       WMB  \
Date                                                                     
2011-01-04  0.004700  0.015762  0.007382  0.012936  0.007492  0.014149   
2011-01-05  0.004768  0.015861  0.007439  0.012991  0.007343  0.014147   
2011-01-06  0.004751  0.015846  0.007445  0.012999  0.007362  0.014128   
2011-01-07  0.004765  0.015847  0.007441  0.013006  0.007361  0.014116   
2011-01-10  0.004748  0.015876  0.007452  0.012994  0.007363  0.014120   

                FITB      CHRW  
Date                            
2011-01-04  0.017356  0.008058  
2011-01-05  0.017316  0.008082  
2011-01-06  0.017306  0.008083  
2011-01-07  0.017313  0.008089  
2011-01-10  0.017248  0.008104

### 4.2 Estimating Size Betas

We estimate stocks's sensitivity to the size on daily data using linear regression (implemented in the size_betas function from computation_functions.py script).

In [15]:
size_betas = cf.size_betas(excess_returns, size)
size_betas.head()

stock:TER, row: 0
stock:TER, row: 1
stock:TER, row: 2
stock:TER, row: 3
stock:TER, row: 4
stock:TER, row: 5
stock:TER, row: 6
stock:TER, row: 7
stock:TER, row: 8
stock:TER, row: 9
stock:TER, row: 10
stock:TER, row: 11
stock:TER, row: 12
stock:TER, row: 13
stock:TER, row: 14
stock:TER, row: 15
stock:TER, row: 16
stock:TER, row: 17
stock:TER, row: 18
stock:TER, row: 19
stock:TER, row: 20
stock:TER, row: 21
stock:TER, row: 22
stock:TER, row: 23
stock:TER, row: 24
stock:TER, row: 25
stock:TER, row: 26
stock:TER, row: 27
stock:TER, row: 28
stock:TER, row: 29
stock:TER, row: 30
stock:TER, row: 31
stock:TER, row: 32
stock:TER, row: 33
stock:TER, row: 34
stock:TER, row: 35
stock:TER, row: 36
stock:TER, row: 37
stock:TER, row: 38
stock:TER, row: 39
stock:TER, row: 40
stock:TER, row: 41
stock:TER, row: 42
stock:TER, row: 43
stock:TER, row: 44
stock:TER, row: 45
stock:TER, row: 46
stock:TER, row: 47
stock:TER, row: 48
stock:TER, row: 49
stock:TER, row: 50
stock:TER, row: 51
stock:TER, row: 52
sto

stock:TER, row: 941
stock:TER, row: 942
stock:TER, row: 943
stock:TER, row: 944
stock:TER, row: 945
stock:TER, row: 946
stock:TER, row: 947
stock:TER, row: 948
stock:TER, row: 949
stock:TER, row: 950
stock:TER, row: 951
stock:TER, row: 952
stock:TER, row: 953
stock:TER, row: 954
stock:TER, row: 955
stock:TER, row: 956
stock:TER, row: 957
stock:TER, row: 958
stock:TER, row: 959
stock:TER, row: 960
stock:TER, row: 961
stock:TER, row: 962
stock:TER, row: 963
stock:TER, row: 964
stock:TER, row: 965
stock:TER, row: 966
stock:TER, row: 967
stock:TER, row: 968
stock:TER, row: 969
stock:TER, row: 970
stock:TER, row: 971
stock:TER, row: 972
stock:TER, row: 973
stock:TER, row: 974
stock:TER, row: 975
stock:TER, row: 976
stock:TER, row: 977
stock:TER, row: 978
stock:TER, row: 979
stock:TER, row: 980
stock:TER, row: 981
stock:TER, row: 982
stock:TER, row: 983
stock:TER, row: 984
stock:TER, row: 985
stock:TER, row: 986
stock:TER, row: 987
stock:TER, row: 988
stock:TER, row: 989
stock:TER, row: 990


stock:TER, row: 1587
stock:TER, row: 1588
stock:TER, row: 1589
stock:TER, row: 1590
stock:TER, row: 1591
stock:TER, row: 1592
stock:TER, row: 1593
stock:TER, row: 1594
stock:TER, row: 1595
stock:TER, row: 1596
stock:TER, row: 1597
stock:TER, row: 1598
stock:TER, row: 1599
stock:TER, row: 1600
stock:TER, row: 1601
stock:TER, row: 1602
stock:TER, row: 1603
stock:TER, row: 1604
stock:TER, row: 1605
stock:TER, row: 1606
stock:TER, row: 1607
stock:TER, row: 1608
stock:TER, row: 1609
stock:TER, row: 1610
stock:TER, row: 1611
stock:TER, row: 1612
stock:TER, row: 1613
stock:TER, row: 1614
stock:TER, row: 1615
stock:TER, row: 1616
stock:TER, row: 1617
stock:TER, row: 1618
stock:TER, row: 1619
stock:TER, row: 1620
stock:TER, row: 1621
stock:TER, row: 1622
stock:TER, row: 1623
stock:TER, row: 1624
stock:TER, row: 1625
stock:TER, row: 1626
stock:TER, row: 1627
stock:TER, row: 1628
stock:TER, row: 1629
stock:TER, row: 1630
stock:TER, row: 1631
stock:TER, row: 1632
stock:TER, row: 1633
stock:TER, ro

stock:TER, row: 2430
stock:TER, row: 2431
stock:TER, row: 2432
stock:TER, row: 2433
stock:TER, row: 2434
stock:TER, row: 2435
stock:TER, row: 2436
stock:TER, row: 2437
stock:TER, row: 2438
stock:TER, row: 2439
stock:TER, row: 2440
stock:TER, row: 2441
stock:TER, row: 2442
stock:TER, row: 2443
stock:TER, row: 2444
stock:TER, row: 2445
stock:TER, row: 2446
stock:TER, row: 2447
stock:TER, row: 2448
stock:TER, row: 2449
stock:TER, row: 2450
stock:TER, row: 2451
stock:TER, row: 2452
stock:TER, row: 2453
stock:TER, row: 2454
stock:TER, row: 2455
stock:TER, row: 2456
stock:TER, row: 2457
stock:TER, row: 2458
stock:TER, row: 2459
stock:TER, row: 2460
stock:TER, row: 2461
stock:TER, row: 2462
stock:TER, row: 2463
stock:TER, row: 2464
stock:TER, row: 2465
stock:TER, row: 2466
stock:TER, row: 2467
stock:TER, row: 2468
stock:TER, row: 2469
stock:TER, row: 2470
stock:TER, row: 2471
stock:TER, row: 2472
stock:TER, row: 2473
stock:TER, row: 2474
stock:TER, row: 2475
stock:TER, row: 2476
stock:TER, ro

stock:ADM, row: 133
stock:ADM, row: 134
stock:ADM, row: 135
stock:ADM, row: 136
stock:ADM, row: 137
stock:ADM, row: 138
stock:ADM, row: 139
stock:ADM, row: 140
stock:ADM, row: 141
stock:ADM, row: 142
stock:ADM, row: 143
stock:ADM, row: 144
stock:ADM, row: 145
stock:ADM, row: 146
stock:ADM, row: 147
stock:ADM, row: 148
stock:ADM, row: 149
stock:ADM, row: 150
stock:ADM, row: 151
stock:ADM, row: 152
stock:ADM, row: 153
stock:ADM, row: 154
stock:ADM, row: 155
stock:ADM, row: 156
stock:ADM, row: 157
stock:ADM, row: 158
stock:ADM, row: 159
stock:ADM, row: 160
stock:ADM, row: 161
stock:ADM, row: 162
stock:ADM, row: 163
stock:ADM, row: 164
stock:ADM, row: 165
stock:ADM, row: 166
stock:ADM, row: 167
stock:ADM, row: 168
stock:ADM, row: 169
stock:ADM, row: 170
stock:ADM, row: 171
stock:ADM, row: 172
stock:ADM, row: 173
stock:ADM, row: 174
stock:ADM, row: 175
stock:ADM, row: 176
stock:ADM, row: 177
stock:ADM, row: 178
stock:ADM, row: 179
stock:ADM, row: 180
stock:ADM, row: 181
stock:ADM, row: 182


stock:ADM, row: 1088
stock:ADM, row: 1089
stock:ADM, row: 1090
stock:ADM, row: 1091
stock:ADM, row: 1092
stock:ADM, row: 1093
stock:ADM, row: 1094
stock:ADM, row: 1095
stock:ADM, row: 1096
stock:ADM, row: 1097
stock:ADM, row: 1098
stock:ADM, row: 1099
stock:ADM, row: 1100
stock:ADM, row: 1101
stock:ADM, row: 1102
stock:ADM, row: 1103
stock:ADM, row: 1104
stock:ADM, row: 1105
stock:ADM, row: 1106
stock:ADM, row: 1107
stock:ADM, row: 1108
stock:ADM, row: 1109
stock:ADM, row: 1110
stock:ADM, row: 1111
stock:ADM, row: 1112
stock:ADM, row: 1113
stock:ADM, row: 1114
stock:ADM, row: 1115
stock:ADM, row: 1116
stock:ADM, row: 1117
stock:ADM, row: 1118
stock:ADM, row: 1119
stock:ADM, row: 1120
stock:ADM, row: 1121
stock:ADM, row: 1122
stock:ADM, row: 1123
stock:ADM, row: 1124
stock:ADM, row: 1125
stock:ADM, row: 1126
stock:ADM, row: 1127
stock:ADM, row: 1128
stock:ADM, row: 1129
stock:ADM, row: 1130
stock:ADM, row: 1131
stock:ADM, row: 1132
stock:ADM, row: 1133
stock:ADM, row: 1134
stock:ADM, ro

stock:ADM, row: 1831
stock:ADM, row: 1832
stock:ADM, row: 1833
stock:ADM, row: 1834
stock:ADM, row: 1835
stock:ADM, row: 1836
stock:ADM, row: 1837
stock:ADM, row: 1838
stock:ADM, row: 1839
stock:ADM, row: 1840
stock:ADM, row: 1841
stock:ADM, row: 1842
stock:ADM, row: 1843
stock:ADM, row: 1844
stock:ADM, row: 1845
stock:ADM, row: 1846
stock:ADM, row: 1847
stock:ADM, row: 1848
stock:ADM, row: 1849
stock:ADM, row: 1850
stock:ADM, row: 1851
stock:ADM, row: 1852
stock:ADM, row: 1853
stock:ADM, row: 1854
stock:ADM, row: 1855
stock:ADM, row: 1856
stock:ADM, row: 1857
stock:ADM, row: 1858
stock:ADM, row: 1859
stock:ADM, row: 1860
stock:ADM, row: 1861
stock:ADM, row: 1862
stock:ADM, row: 1863
stock:ADM, row: 1864
stock:ADM, row: 1865
stock:ADM, row: 1866
stock:ADM, row: 1867
stock:ADM, row: 1868
stock:ADM, row: 1869
stock:ADM, row: 1870
stock:ADM, row: 1871
stock:ADM, row: 1872
stock:ADM, row: 1873
stock:ADM, row: 1874
stock:ADM, row: 1875
stock:ADM, row: 1876
stock:ADM, row: 1877
stock:ADM, ro

stock:ADM, row: 2814
stock:ADM, row: 2815
stock:ADM, row: 2816
stock:ADM, row: 2817
stock:ADM, row: 2818
stock:ADM, row: 2819
stock:ADM, row: 2820
stock:ADM, row: 2821
stock:ADM, row: 2822
stock:ADM, row: 2823
stock:ADM, row: 2824
stock:ADM, row: 2825
stock:ADM, row: 2826
stock:ADM, row: 2827
stock:ADM, row: 2828
stock:ADM, row: 2829
stock:ADM, row: 2830
stock:ADM, row: 2831
stock:ADM, row: 2832
stock:ADM, row: 2833
stock:ADM, row: 2834
stock:ADM, row: 2835
stock:ADM, row: 2836
stock:ADM, row: 2837
stock:ADM, row: 2838
stock:ADM, row: 2839
stock:ADM, row: 2840
stock:ADM, row: 2841
stock:ADM, row: 2842
stock:ADM, row: 2843
stock:ADM, row: 2844
stock:ADM, row: 2845
stock:ADM, row: 2846
stock:ADM, row: 2847
stock:ADM, row: 2848
stock:ADM, row: 2849
stock:ADM, row: 2850
stock:ADM, row: 2851
stock:ADM, row: 2852
stock:ADM, row: 2853
stock:ADM, row: 2854
stock:ADM, row: 2855
stock:ADM, row: 2856
stock:ADM, row: 2857
stock:ADM, row: 2858
stock:ADM, row: 2859
stock:ADM, row: 2860
stock:ADM, ro

stock:TMUS, row: 264
stock:TMUS, row: 265
stock:TMUS, row: 266
stock:TMUS, row: 267
stock:TMUS, row: 268
stock:TMUS, row: 269
stock:TMUS, row: 270
stock:TMUS, row: 271
stock:TMUS, row: 272
stock:TMUS, row: 273
stock:TMUS, row: 274
stock:TMUS, row: 275
stock:TMUS, row: 276
stock:TMUS, row: 277
stock:TMUS, row: 278
stock:TMUS, row: 279
stock:TMUS, row: 280
stock:TMUS, row: 281
stock:TMUS, row: 282
stock:TMUS, row: 283
stock:TMUS, row: 284
stock:TMUS, row: 285
stock:TMUS, row: 286
stock:TMUS, row: 287
stock:TMUS, row: 288
stock:TMUS, row: 289
stock:TMUS, row: 290
stock:TMUS, row: 291
stock:TMUS, row: 292
stock:TMUS, row: 293
stock:TMUS, row: 294
stock:TMUS, row: 295
stock:TMUS, row: 296
stock:TMUS, row: 297
stock:TMUS, row: 298
stock:TMUS, row: 299
stock:TMUS, row: 300
stock:TMUS, row: 301
stock:TMUS, row: 302
stock:TMUS, row: 303
stock:TMUS, row: 304
stock:TMUS, row: 305
stock:TMUS, row: 306
stock:TMUS, row: 307
stock:TMUS, row: 308
stock:TMUS, row: 309
stock:TMUS, row: 310
stock:TMUS, r

stock:TMUS, row: 1292
stock:TMUS, row: 1293
stock:TMUS, row: 1294
stock:TMUS, row: 1295
stock:TMUS, row: 1296
stock:TMUS, row: 1297
stock:TMUS, row: 1298
stock:TMUS, row: 1299
stock:TMUS, row: 1300
stock:TMUS, row: 1301
stock:TMUS, row: 1302
stock:TMUS, row: 1303
stock:TMUS, row: 1304
stock:TMUS, row: 1305
stock:TMUS, row: 1306
stock:TMUS, row: 1307
stock:TMUS, row: 1308
stock:TMUS, row: 1309
stock:TMUS, row: 1310
stock:TMUS, row: 1311
stock:TMUS, row: 1312
stock:TMUS, row: 1313
stock:TMUS, row: 1314
stock:TMUS, row: 1315
stock:TMUS, row: 1316
stock:TMUS, row: 1317
stock:TMUS, row: 1318
stock:TMUS, row: 1319
stock:TMUS, row: 1320
stock:TMUS, row: 1321
stock:TMUS, row: 1322
stock:TMUS, row: 1323
stock:TMUS, row: 1324
stock:TMUS, row: 1325
stock:TMUS, row: 1326
stock:TMUS, row: 1327
stock:TMUS, row: 1328
stock:TMUS, row: 1329
stock:TMUS, row: 1330
stock:TMUS, row: 1331
stock:TMUS, row: 1332
stock:TMUS, row: 1333
stock:TMUS, row: 1334
stock:TMUS, row: 1335
stock:TMUS, row: 1336
stock:TMUS

stock:TMUS, row: 1955
stock:TMUS, row: 1956
stock:TMUS, row: 1957
stock:TMUS, row: 1958
stock:TMUS, row: 1959
stock:TMUS, row: 1960
stock:TMUS, row: 1961
stock:TMUS, row: 1962
stock:TMUS, row: 1963
stock:TMUS, row: 1964
stock:TMUS, row: 1965
stock:TMUS, row: 1966
stock:TMUS, row: 1967
stock:TMUS, row: 1968
stock:TMUS, row: 1969
stock:TMUS, row: 1970
stock:TMUS, row: 1971
stock:TMUS, row: 1972
stock:TMUS, row: 1973
stock:TMUS, row: 1974
stock:TMUS, row: 1975
stock:TMUS, row: 1976
stock:TMUS, row: 1977
stock:TMUS, row: 1978
stock:TMUS, row: 1979
stock:TMUS, row: 1980
stock:TMUS, row: 1981
stock:TMUS, row: 1982
stock:TMUS, row: 1983
stock:TMUS, row: 1984
stock:TMUS, row: 1985
stock:TMUS, row: 1986
stock:TMUS, row: 1987
stock:TMUS, row: 1988
stock:TMUS, row: 1989
stock:TMUS, row: 1990
stock:TMUS, row: 1991
stock:TMUS, row: 1992
stock:TMUS, row: 1993
stock:TMUS, row: 1994
stock:TMUS, row: 1995
stock:TMUS, row: 1996
stock:TMUS, row: 1997
stock:TMUS, row: 1998
stock:TMUS, row: 1999
stock:TMUS

stock:TMUS, row: 2716
stock:TMUS, row: 2717
stock:TMUS, row: 2718
stock:TMUS, row: 2719
stock:TMUS, row: 2720
stock:TMUS, row: 2721
stock:TMUS, row: 2722
stock:TMUS, row: 2723
stock:TMUS, row: 2724
stock:TMUS, row: 2725
stock:TMUS, row: 2726
stock:TMUS, row: 2727
stock:TMUS, row: 2728
stock:TMUS, row: 2729
stock:TMUS, row: 2730
stock:TMUS, row: 2731
stock:TMUS, row: 2732
stock:TMUS, row: 2733
stock:TMUS, row: 2734
stock:TMUS, row: 2735
stock:TMUS, row: 2736
stock:TMUS, row: 2737
stock:TMUS, row: 2738
stock:TMUS, row: 2739
stock:TMUS, row: 2740
stock:TMUS, row: 2741
stock:TMUS, row: 2742
stock:TMUS, row: 2743
stock:TMUS, row: 2744
stock:TMUS, row: 2745
stock:TMUS, row: 2746
stock:TMUS, row: 2747
stock:TMUS, row: 2748
stock:TMUS, row: 2749
stock:TMUS, row: 2750
stock:TMUS, row: 2751
stock:TMUS, row: 2752
stock:TMUS, row: 2753
stock:TMUS, row: 2754
stock:TMUS, row: 2755
stock:TMUS, row: 2756
stock:TMUS, row: 2757
stock:TMUS, row: 2758
stock:TMUS, row: 2759
stock:TMUS, row: 2760
stock:TMUS

stock:IDXX, row: 762
stock:IDXX, row: 763
stock:IDXX, row: 764
stock:IDXX, row: 765
stock:IDXX, row: 766
stock:IDXX, row: 767
stock:IDXX, row: 768
stock:IDXX, row: 769
stock:IDXX, row: 770
stock:IDXX, row: 771
stock:IDXX, row: 772
stock:IDXX, row: 773
stock:IDXX, row: 774
stock:IDXX, row: 775
stock:IDXX, row: 776
stock:IDXX, row: 777
stock:IDXX, row: 778
stock:IDXX, row: 779
stock:IDXX, row: 780
stock:IDXX, row: 781
stock:IDXX, row: 782
stock:IDXX, row: 783
stock:IDXX, row: 784
stock:IDXX, row: 785
stock:IDXX, row: 786
stock:IDXX, row: 787
stock:IDXX, row: 788
stock:IDXX, row: 789
stock:IDXX, row: 790
stock:IDXX, row: 791
stock:IDXX, row: 792
stock:IDXX, row: 793
stock:IDXX, row: 794
stock:IDXX, row: 795
stock:IDXX, row: 796
stock:IDXX, row: 797
stock:IDXX, row: 798
stock:IDXX, row: 799
stock:IDXX, row: 800
stock:IDXX, row: 801
stock:IDXX, row: 802
stock:IDXX, row: 803
stock:IDXX, row: 804
stock:IDXX, row: 805
stock:IDXX, row: 806
stock:IDXX, row: 807
stock:IDXX, row: 808
stock:IDXX, r

stock:IDXX, row: 1242
stock:IDXX, row: 1243
stock:IDXX, row: 1244
stock:IDXX, row: 1245
stock:IDXX, row: 1246
stock:IDXX, row: 1247
stock:IDXX, row: 1248
stock:IDXX, row: 1249
stock:IDXX, row: 1250
stock:IDXX, row: 1251
stock:IDXX, row: 1252
stock:IDXX, row: 1253
stock:IDXX, row: 1254
stock:IDXX, row: 1255
stock:IDXX, row: 1256
stock:IDXX, row: 1257
stock:IDXX, row: 1258
stock:IDXX, row: 1259
stock:IDXX, row: 1260
stock:IDXX, row: 1261
stock:IDXX, row: 1262
stock:IDXX, row: 1263
stock:IDXX, row: 1264
stock:IDXX, row: 1265
stock:IDXX, row: 1266
stock:IDXX, row: 1267
stock:IDXX, row: 1268
stock:IDXX, row: 1269
stock:IDXX, row: 1270
stock:IDXX, row: 1271
stock:IDXX, row: 1272
stock:IDXX, row: 1273
stock:IDXX, row: 1274
stock:IDXX, row: 1275
stock:IDXX, row: 1276
stock:IDXX, row: 1277
stock:IDXX, row: 1278
stock:IDXX, row: 1279
stock:IDXX, row: 1280
stock:IDXX, row: 1281
stock:IDXX, row: 1282
stock:IDXX, row: 1283
stock:IDXX, row: 1284
stock:IDXX, row: 1285
stock:IDXX, row: 1286
stock:IDXX

stock:IDXX, row: 2175
stock:IDXX, row: 2176
stock:IDXX, row: 2177
stock:IDXX, row: 2178
stock:IDXX, row: 2179
stock:IDXX, row: 2180
stock:IDXX, row: 2181
stock:IDXX, row: 2182
stock:IDXX, row: 2183
stock:IDXX, row: 2184
stock:IDXX, row: 2185
stock:IDXX, row: 2186
stock:IDXX, row: 2187
stock:IDXX, row: 2188
stock:IDXX, row: 2189
stock:IDXX, row: 2190
stock:IDXX, row: 2191
stock:IDXX, row: 2192
stock:IDXX, row: 2193
stock:IDXX, row: 2194
stock:IDXX, row: 2195
stock:IDXX, row: 2196
stock:IDXX, row: 2197
stock:IDXX, row: 2198
stock:IDXX, row: 2199
stock:IDXX, row: 2200
stock:IDXX, row: 2201
stock:IDXX, row: 2202
stock:IDXX, row: 2203
stock:IDXX, row: 2204
stock:IDXX, row: 2205
stock:IDXX, row: 2206
stock:IDXX, row: 2207
stock:IDXX, row: 2208
stock:IDXX, row: 2209
stock:IDXX, row: 2210
stock:IDXX, row: 2211
stock:IDXX, row: 2212
stock:IDXX, row: 2213
stock:IDXX, row: 2214
stock:IDXX, row: 2215
stock:IDXX, row: 2216
stock:IDXX, row: 2217
stock:IDXX, row: 2218
stock:IDXX, row: 2219
stock:IDXX

stock:IDXX, row: 2577
stock:IDXX, row: 2578
stock:IDXX, row: 2579
stock:IDXX, row: 2580
stock:IDXX, row: 2581
stock:IDXX, row: 2582
stock:IDXX, row: 2583
stock:IDXX, row: 2584
stock:IDXX, row: 2585
stock:IDXX, row: 2586
stock:IDXX, row: 2587
stock:IDXX, row: 2588
stock:IDXX, row: 2589
stock:IDXX, row: 2590
stock:IDXX, row: 2591
stock:IDXX, row: 2592
stock:IDXX, row: 2593
stock:IDXX, row: 2594
stock:IDXX, row: 2595
stock:IDXX, row: 2596
stock:IDXX, row: 2597
stock:IDXX, row: 2598
stock:IDXX, row: 2599
stock:IDXX, row: 2600
stock:IDXX, row: 2601
stock:IDXX, row: 2602
stock:IDXX, row: 2603
stock:IDXX, row: 2604
stock:IDXX, row: 2605
stock:IDXX, row: 2606
stock:IDXX, row: 2607
stock:IDXX, row: 2608
stock:IDXX, row: 2609
stock:IDXX, row: 2610
stock:IDXX, row: 2611
stock:IDXX, row: 2612
stock:IDXX, row: 2613
stock:IDXX, row: 2614
stock:IDXX, row: 2615
stock:IDXX, row: 2616
stock:IDXX, row: 2617
stock:IDXX, row: 2618
stock:IDXX, row: 2619
stock:IDXX, row: 2620
stock:IDXX, row: 2621
stock:IDXX

stock:LNC, row: 222
stock:LNC, row: 223
stock:LNC, row: 224
stock:LNC, row: 225
stock:LNC, row: 226
stock:LNC, row: 227
stock:LNC, row: 228
stock:LNC, row: 229
stock:LNC, row: 230
stock:LNC, row: 231
stock:LNC, row: 232
stock:LNC, row: 233
stock:LNC, row: 234
stock:LNC, row: 235
stock:LNC, row: 236
stock:LNC, row: 237
stock:LNC, row: 238
stock:LNC, row: 239
stock:LNC, row: 240
stock:LNC, row: 241
stock:LNC, row: 242
stock:LNC, row: 243
stock:LNC, row: 244
stock:LNC, row: 245
stock:LNC, row: 246
stock:LNC, row: 247
stock:LNC, row: 248
stock:LNC, row: 249
stock:LNC, row: 250
stock:LNC, row: 251
stock:LNC, row: 252
stock:LNC, row: 253
stock:LNC, row: 254
stock:LNC, row: 255
stock:LNC, row: 256
stock:LNC, row: 257
stock:LNC, row: 258
stock:LNC, row: 259
stock:LNC, row: 260
stock:LNC, row: 261
stock:LNC, row: 262
stock:LNC, row: 263
stock:LNC, row: 264
stock:LNC, row: 265
stock:LNC, row: 266
stock:LNC, row: 267
stock:LNC, row: 268
stock:LNC, row: 269
stock:LNC, row: 270
stock:LNC, row: 271


stock:LNC, row: 715
stock:LNC, row: 716
stock:LNC, row: 717
stock:LNC, row: 718
stock:LNC, row: 719
stock:LNC, row: 720
stock:LNC, row: 721
stock:LNC, row: 722
stock:LNC, row: 723
stock:LNC, row: 724
stock:LNC, row: 725
stock:LNC, row: 726
stock:LNC, row: 727
stock:LNC, row: 728
stock:LNC, row: 729
stock:LNC, row: 730
stock:LNC, row: 731
stock:LNC, row: 732
stock:LNC, row: 733
stock:LNC, row: 734
stock:LNC, row: 735
stock:LNC, row: 736
stock:LNC, row: 737
stock:LNC, row: 738
stock:LNC, row: 739
stock:LNC, row: 740
stock:LNC, row: 741
stock:LNC, row: 742
stock:LNC, row: 743
stock:LNC, row: 744
stock:LNC, row: 745
stock:LNC, row: 746
stock:LNC, row: 747
stock:LNC, row: 748
stock:LNC, row: 749
stock:LNC, row: 750
stock:LNC, row: 751
stock:LNC, row: 752
stock:LNC, row: 753
stock:LNC, row: 754
stock:LNC, row: 755
stock:LNC, row: 756
stock:LNC, row: 757
stock:LNC, row: 758
stock:LNC, row: 759
stock:LNC, row: 760
stock:LNC, row: 761
stock:LNC, row: 762
stock:LNC, row: 763
stock:LNC, row: 764


stock:LNC, row: 1684
stock:LNC, row: 1685
stock:LNC, row: 1686
stock:LNC, row: 1687
stock:LNC, row: 1688
stock:LNC, row: 1689
stock:LNC, row: 1690
stock:LNC, row: 1691
stock:LNC, row: 1692
stock:LNC, row: 1693
stock:LNC, row: 1694
stock:LNC, row: 1695
stock:LNC, row: 1696
stock:LNC, row: 1697
stock:LNC, row: 1698
stock:LNC, row: 1699
stock:LNC, row: 1700
stock:LNC, row: 1701
stock:LNC, row: 1702
stock:LNC, row: 1703
stock:LNC, row: 1704
stock:LNC, row: 1705
stock:LNC, row: 1706
stock:LNC, row: 1707
stock:LNC, row: 1708
stock:LNC, row: 1709
stock:LNC, row: 1710
stock:LNC, row: 1711
stock:LNC, row: 1712
stock:LNC, row: 1713
stock:LNC, row: 1714
stock:LNC, row: 1715
stock:LNC, row: 1716
stock:LNC, row: 1717
stock:LNC, row: 1718
stock:LNC, row: 1719
stock:LNC, row: 1720
stock:LNC, row: 1721
stock:LNC, row: 1722
stock:LNC, row: 1723
stock:LNC, row: 1724
stock:LNC, row: 1725
stock:LNC, row: 1726
stock:LNC, row: 1727
stock:LNC, row: 1728
stock:LNC, row: 1729
stock:LNC, row: 1730
stock:LNC, ro

stock:LNC, row: 2945
stock:LNC, row: 2946
stock:LNC, row: 2947
stock:LNC, row: 2948
stock:LNC, row: 2949
stock:LNC, row: 2950
stock:LNC, row: 2951
stock:LNC, row: 2952
stock:LNC, row: 2953
stock:LNC, row: 2954
stock:LNC, row: 2955
stock:LNC, row: 2956
stock:LNC, row: 2957
stock:LNC, row: 2958
stock:LNC, row: 2959
stock:LNC, row: 2960
stock:LNC, row: 2961
stock:LNC, row: 2962
stock:LNC, row: 2963
stock:LNC, row: 2964
stock:LNC, row: 2965
stock:LNC, row: 2966
stock:LNC, row: 2967
stock:LNC, row: 2968
stock:LNC, row: 2969
stock:LNC, row: 2970
stock:LNC, row: 2971
stock:LNC, row: 2972
stock:LNC, row: 2973
stock:LNC, row: 2974
stock:LNC, row: 2975
stock:LNC, row: 2976
stock:WST, row: 0
stock:WST, row: 1
stock:WST, row: 2
stock:WST, row: 3
stock:WST, row: 4
stock:WST, row: 5
stock:WST, row: 6
stock:WST, row: 7
stock:WST, row: 8
stock:WST, row: 9
stock:WST, row: 10
stock:WST, row: 11
stock:WST, row: 12
stock:WST, row: 13
stock:WST, row: 14
stock:WST, row: 15
stock:WST, row: 16
stock:WST, row:

stock:WST, row: 1273
stock:WST, row: 1274
stock:WST, row: 1275
stock:WST, row: 1276
stock:WST, row: 1277
stock:WST, row: 1278
stock:WST, row: 1279
stock:WST, row: 1280
stock:WST, row: 1281
stock:WST, row: 1282
stock:WST, row: 1283
stock:WST, row: 1284
stock:WST, row: 1285
stock:WST, row: 1286
stock:WST, row: 1287
stock:WST, row: 1288
stock:WST, row: 1289
stock:WST, row: 1290
stock:WST, row: 1291
stock:WST, row: 1292
stock:WST, row: 1293
stock:WST, row: 1294
stock:WST, row: 1295
stock:WST, row: 1296
stock:WST, row: 1297
stock:WST, row: 1298
stock:WST, row: 1299
stock:WST, row: 1300
stock:WST, row: 1301
stock:WST, row: 1302
stock:WST, row: 1303
stock:WST, row: 1304
stock:WST, row: 1305
stock:WST, row: 1306
stock:WST, row: 1307
stock:WST, row: 1308
stock:WST, row: 1309
stock:WST, row: 1310
stock:WST, row: 1311
stock:WST, row: 1312
stock:WST, row: 1313
stock:WST, row: 1314
stock:WST, row: 1315
stock:WST, row: 1316
stock:WST, row: 1317
stock:WST, row: 1318
stock:WST, row: 1319
stock:WST, ro

stock:WST, row: 1931
stock:WST, row: 1932
stock:WST, row: 1933
stock:WST, row: 1934
stock:WST, row: 1935
stock:WST, row: 1936
stock:WST, row: 1937
stock:WST, row: 1938
stock:WST, row: 1939
stock:WST, row: 1940
stock:WST, row: 1941
stock:WST, row: 1942
stock:WST, row: 1943
stock:WST, row: 1944
stock:WST, row: 1945
stock:WST, row: 1946
stock:WST, row: 1947
stock:WST, row: 1948
stock:WST, row: 1949
stock:WST, row: 1950
stock:WST, row: 1951
stock:WST, row: 1952
stock:WST, row: 1953
stock:WST, row: 1954
stock:WST, row: 1955
stock:WST, row: 1956
stock:WST, row: 1957
stock:WST, row: 1958
stock:WST, row: 1959
stock:WST, row: 1960
stock:WST, row: 1961
stock:WST, row: 1962
stock:WST, row: 1963
stock:WST, row: 1964
stock:WST, row: 1965
stock:WST, row: 1966
stock:WST, row: 1967
stock:WST, row: 1968
stock:WST, row: 1969
stock:WST, row: 1970
stock:WST, row: 1971
stock:WST, row: 1972
stock:WST, row: 1973
stock:WST, row: 1974
stock:WST, row: 1975
stock:WST, row: 1976
stock:WST, row: 1977
stock:WST, ro

stock:WST, row: 2722
stock:WST, row: 2723
stock:WST, row: 2724
stock:WST, row: 2725
stock:WST, row: 2726
stock:WST, row: 2727
stock:WST, row: 2728
stock:WST, row: 2729
stock:WST, row: 2730
stock:WST, row: 2731
stock:WST, row: 2732
stock:WST, row: 2733
stock:WST, row: 2734
stock:WST, row: 2735
stock:WST, row: 2736
stock:WST, row: 2737
stock:WST, row: 2738
stock:WST, row: 2739
stock:WST, row: 2740
stock:WST, row: 2741
stock:WST, row: 2742
stock:WST, row: 2743
stock:WST, row: 2744
stock:WST, row: 2745
stock:WST, row: 2746
stock:WST, row: 2747
stock:WST, row: 2748
stock:WST, row: 2749
stock:WST, row: 2750
stock:WST, row: 2751
stock:WST, row: 2752
stock:WST, row: 2753
stock:WST, row: 2754
stock:WST, row: 2755
stock:WST, row: 2756
stock:WST, row: 2757
stock:WST, row: 2758
stock:WST, row: 2759
stock:WST, row: 2760
stock:WST, row: 2761
stock:WST, row: 2762
stock:WST, row: 2763
stock:WST, row: 2764
stock:WST, row: 2765
stock:WST, row: 2766
stock:WST, row: 2767
stock:WST, row: 2768
stock:WST, ro

stock:AMZN, row: 443
stock:AMZN, row: 444
stock:AMZN, row: 445
stock:AMZN, row: 446
stock:AMZN, row: 447
stock:AMZN, row: 448
stock:AMZN, row: 449
stock:AMZN, row: 450
stock:AMZN, row: 451
stock:AMZN, row: 452
stock:AMZN, row: 453
stock:AMZN, row: 454
stock:AMZN, row: 455
stock:AMZN, row: 456
stock:AMZN, row: 457
stock:AMZN, row: 458
stock:AMZN, row: 459
stock:AMZN, row: 460
stock:AMZN, row: 461
stock:AMZN, row: 462
stock:AMZN, row: 463
stock:AMZN, row: 464
stock:AMZN, row: 465
stock:AMZN, row: 466
stock:AMZN, row: 467
stock:AMZN, row: 468
stock:AMZN, row: 469
stock:AMZN, row: 470
stock:AMZN, row: 471
stock:AMZN, row: 472
stock:AMZN, row: 473
stock:AMZN, row: 474
stock:AMZN, row: 475
stock:AMZN, row: 476
stock:AMZN, row: 477
stock:AMZN, row: 478
stock:AMZN, row: 479
stock:AMZN, row: 480
stock:AMZN, row: 481
stock:AMZN, row: 482
stock:AMZN, row: 483
stock:AMZN, row: 484
stock:AMZN, row: 485
stock:AMZN, row: 486
stock:AMZN, row: 487
stock:AMZN, row: 488
stock:AMZN, row: 489
stock:AMZN, r

stock:AMZN, row: 1307
stock:AMZN, row: 1308
stock:AMZN, row: 1309
stock:AMZN, row: 1310
stock:AMZN, row: 1311
stock:AMZN, row: 1312
stock:AMZN, row: 1313
stock:AMZN, row: 1314
stock:AMZN, row: 1315
stock:AMZN, row: 1316
stock:AMZN, row: 1317
stock:AMZN, row: 1318
stock:AMZN, row: 1319
stock:AMZN, row: 1320
stock:AMZN, row: 1321
stock:AMZN, row: 1322
stock:AMZN, row: 1323
stock:AMZN, row: 1324
stock:AMZN, row: 1325
stock:AMZN, row: 1326
stock:AMZN, row: 1327
stock:AMZN, row: 1328
stock:AMZN, row: 1329
stock:AMZN, row: 1330
stock:AMZN, row: 1331
stock:AMZN, row: 1332
stock:AMZN, row: 1333
stock:AMZN, row: 1334
stock:AMZN, row: 1335
stock:AMZN, row: 1336
stock:AMZN, row: 1337
stock:AMZN, row: 1338
stock:AMZN, row: 1339
stock:AMZN, row: 1340
stock:AMZN, row: 1341
stock:AMZN, row: 1342
stock:AMZN, row: 1343
stock:AMZN, row: 1344
stock:AMZN, row: 1345
stock:AMZN, row: 1346
stock:AMZN, row: 1347
stock:AMZN, row: 1348
stock:AMZN, row: 1349
stock:AMZN, row: 1350
stock:AMZN, row: 1351
stock:AMZN

stock:AMZN, row: 2374
stock:AMZN, row: 2375
stock:AMZN, row: 2376
stock:AMZN, row: 2377
stock:AMZN, row: 2378
stock:AMZN, row: 2379
stock:AMZN, row: 2380
stock:AMZN, row: 2381
stock:AMZN, row: 2382
stock:AMZN, row: 2383
stock:AMZN, row: 2384
stock:AMZN, row: 2385
stock:AMZN, row: 2386
stock:AMZN, row: 2387
stock:AMZN, row: 2388
stock:AMZN, row: 2389
stock:AMZN, row: 2390
stock:AMZN, row: 2391
stock:AMZN, row: 2392
stock:AMZN, row: 2393
stock:AMZN, row: 2394
stock:AMZN, row: 2395
stock:AMZN, row: 2396
stock:AMZN, row: 2397
stock:AMZN, row: 2398
stock:AMZN, row: 2399
stock:AMZN, row: 2400
stock:AMZN, row: 2401
stock:AMZN, row: 2402
stock:AMZN, row: 2403
stock:AMZN, row: 2404
stock:AMZN, row: 2405
stock:AMZN, row: 2406
stock:AMZN, row: 2407
stock:AMZN, row: 2408
stock:AMZN, row: 2409
stock:AMZN, row: 2410
stock:AMZN, row: 2411
stock:AMZN, row: 2412
stock:AMZN, row: 2413
stock:AMZN, row: 2414
stock:AMZN, row: 2415
stock:AMZN, row: 2416
stock:AMZN, row: 2417
stock:AMZN, row: 2418
stock:AMZN

stock:DOV, row: 169
stock:DOV, row: 170
stock:DOV, row: 171
stock:DOV, row: 172
stock:DOV, row: 173
stock:DOV, row: 174
stock:DOV, row: 175
stock:DOV, row: 176
stock:DOV, row: 177
stock:DOV, row: 178
stock:DOV, row: 179
stock:DOV, row: 180
stock:DOV, row: 181
stock:DOV, row: 182
stock:DOV, row: 183
stock:DOV, row: 184
stock:DOV, row: 185
stock:DOV, row: 186
stock:DOV, row: 187
stock:DOV, row: 188
stock:DOV, row: 189
stock:DOV, row: 190
stock:DOV, row: 191
stock:DOV, row: 192
stock:DOV, row: 193
stock:DOV, row: 194
stock:DOV, row: 195
stock:DOV, row: 196
stock:DOV, row: 197
stock:DOV, row: 198
stock:DOV, row: 199
stock:DOV, row: 200
stock:DOV, row: 201
stock:DOV, row: 202
stock:DOV, row: 203
stock:DOV, row: 204
stock:DOV, row: 205
stock:DOV, row: 206
stock:DOV, row: 207
stock:DOV, row: 208
stock:DOV, row: 209
stock:DOV, row: 210
stock:DOV, row: 211
stock:DOV, row: 212
stock:DOV, row: 213
stock:DOV, row: 214
stock:DOV, row: 215
stock:DOV, row: 216
stock:DOV, row: 217
stock:DOV, row: 218


stock:DOV, row: 1289
stock:DOV, row: 1290
stock:DOV, row: 1291
stock:DOV, row: 1292
stock:DOV, row: 1293
stock:DOV, row: 1294
stock:DOV, row: 1295
stock:DOV, row: 1296
stock:DOV, row: 1297
stock:DOV, row: 1298
stock:DOV, row: 1299
stock:DOV, row: 1300
stock:DOV, row: 1301
stock:DOV, row: 1302
stock:DOV, row: 1303
stock:DOV, row: 1304
stock:DOV, row: 1305
stock:DOV, row: 1306
stock:DOV, row: 1307
stock:DOV, row: 1308
stock:DOV, row: 1309
stock:DOV, row: 1310
stock:DOV, row: 1311
stock:DOV, row: 1312
stock:DOV, row: 1313
stock:DOV, row: 1314
stock:DOV, row: 1315
stock:DOV, row: 1316
stock:DOV, row: 1317
stock:DOV, row: 1318
stock:DOV, row: 1319
stock:DOV, row: 1320
stock:DOV, row: 1321
stock:DOV, row: 1322
stock:DOV, row: 1323
stock:DOV, row: 1324
stock:DOV, row: 1325
stock:DOV, row: 1326
stock:DOV, row: 1327
stock:DOV, row: 1328
stock:DOV, row: 1329
stock:DOV, row: 1330
stock:DOV, row: 1331
stock:DOV, row: 1332
stock:DOV, row: 1333
stock:DOV, row: 1334
stock:DOV, row: 1335
stock:DOV, ro

stock:DOV, row: 1764
stock:DOV, row: 1765
stock:DOV, row: 1766
stock:DOV, row: 1767
stock:DOV, row: 1768
stock:DOV, row: 1769
stock:DOV, row: 1770
stock:DOV, row: 1771
stock:DOV, row: 1772
stock:DOV, row: 1773
stock:DOV, row: 1774
stock:DOV, row: 1775
stock:DOV, row: 1776
stock:DOV, row: 1777
stock:DOV, row: 1778
stock:DOV, row: 1779
stock:DOV, row: 1780
stock:DOV, row: 1781
stock:DOV, row: 1782
stock:DOV, row: 1783
stock:DOV, row: 1784
stock:DOV, row: 1785
stock:DOV, row: 1786
stock:DOV, row: 1787
stock:DOV, row: 1788
stock:DOV, row: 1789
stock:DOV, row: 1790
stock:DOV, row: 1791
stock:DOV, row: 1792
stock:DOV, row: 1793
stock:DOV, row: 1794
stock:DOV, row: 1795
stock:DOV, row: 1796
stock:DOV, row: 1797
stock:DOV, row: 1798
stock:DOV, row: 1799
stock:DOV, row: 1800
stock:DOV, row: 1801
stock:DOV, row: 1802
stock:DOV, row: 1803
stock:DOV, row: 1804
stock:DOV, row: 1805
stock:DOV, row: 1806
stock:DOV, row: 1807
stock:DOV, row: 1808
stock:DOV, row: 1809
stock:DOV, row: 1810
stock:DOV, ro

stock:DOV, row: 2175
stock:DOV, row: 2176
stock:DOV, row: 2177
stock:DOV, row: 2178
stock:DOV, row: 2179
stock:DOV, row: 2180
stock:DOV, row: 2181
stock:DOV, row: 2182
stock:DOV, row: 2183
stock:DOV, row: 2184
stock:DOV, row: 2185
stock:DOV, row: 2186
stock:DOV, row: 2187
stock:DOV, row: 2188
stock:DOV, row: 2189
stock:DOV, row: 2190
stock:DOV, row: 2191
stock:DOV, row: 2192
stock:DOV, row: 2193
stock:DOV, row: 2194
stock:DOV, row: 2195
stock:DOV, row: 2196
stock:DOV, row: 2197
stock:DOV, row: 2198
stock:DOV, row: 2199
stock:DOV, row: 2200
stock:DOV, row: 2201
stock:DOV, row: 2202
stock:DOV, row: 2203
stock:DOV, row: 2204
stock:DOV, row: 2205
stock:DOV, row: 2206
stock:DOV, row: 2207
stock:DOV, row: 2208
stock:DOV, row: 2209
stock:DOV, row: 2210
stock:DOV, row: 2211
stock:DOV, row: 2212
stock:DOV, row: 2213
stock:DOV, row: 2214
stock:DOV, row: 2215
stock:DOV, row: 2216
stock:DOV, row: 2217
stock:DOV, row: 2218
stock:DOV, row: 2219
stock:DOV, row: 2220
stock:DOV, row: 2221
stock:DOV, ro

stock:DOV, row: 2970
stock:DOV, row: 2971
stock:DOV, row: 2972
stock:DOV, row: 2973
stock:DOV, row: 2974
stock:DOV, row: 2975
stock:DOV, row: 2976
stock:ORLY, row: 0
stock:ORLY, row: 1
stock:ORLY, row: 2
stock:ORLY, row: 3
stock:ORLY, row: 4
stock:ORLY, row: 5
stock:ORLY, row: 6
stock:ORLY, row: 7
stock:ORLY, row: 8
stock:ORLY, row: 9
stock:ORLY, row: 10
stock:ORLY, row: 11
stock:ORLY, row: 12
stock:ORLY, row: 13
stock:ORLY, row: 14
stock:ORLY, row: 15
stock:ORLY, row: 16
stock:ORLY, row: 17
stock:ORLY, row: 18
stock:ORLY, row: 19
stock:ORLY, row: 20
stock:ORLY, row: 21
stock:ORLY, row: 22
stock:ORLY, row: 23
stock:ORLY, row: 24
stock:ORLY, row: 25
stock:ORLY, row: 26
stock:ORLY, row: 27
stock:ORLY, row: 28
stock:ORLY, row: 29
stock:ORLY, row: 30
stock:ORLY, row: 31
stock:ORLY, row: 32
stock:ORLY, row: 33
stock:ORLY, row: 34
stock:ORLY, row: 35
stock:ORLY, row: 36
stock:ORLY, row: 37
stock:ORLY, row: 38
stock:ORLY, row: 39
stock:ORLY, row: 40
stock:ORLY, row: 41
stock:ORLY, row: 42
sto

stock:ORLY, row: 517
stock:ORLY, row: 518
stock:ORLY, row: 519
stock:ORLY, row: 520
stock:ORLY, row: 521
stock:ORLY, row: 522
stock:ORLY, row: 523
stock:ORLY, row: 524
stock:ORLY, row: 525
stock:ORLY, row: 526
stock:ORLY, row: 527
stock:ORLY, row: 528
stock:ORLY, row: 529
stock:ORLY, row: 530
stock:ORLY, row: 531
stock:ORLY, row: 532
stock:ORLY, row: 533
stock:ORLY, row: 534
stock:ORLY, row: 535
stock:ORLY, row: 536
stock:ORLY, row: 537
stock:ORLY, row: 538
stock:ORLY, row: 539
stock:ORLY, row: 540
stock:ORLY, row: 541
stock:ORLY, row: 542
stock:ORLY, row: 543
stock:ORLY, row: 544
stock:ORLY, row: 545
stock:ORLY, row: 546
stock:ORLY, row: 547
stock:ORLY, row: 548
stock:ORLY, row: 549
stock:ORLY, row: 550
stock:ORLY, row: 551
stock:ORLY, row: 552
stock:ORLY, row: 553
stock:ORLY, row: 554
stock:ORLY, row: 555
stock:ORLY, row: 556
stock:ORLY, row: 557
stock:ORLY, row: 558
stock:ORLY, row: 559
stock:ORLY, row: 560
stock:ORLY, row: 561
stock:ORLY, row: 562
stock:ORLY, row: 563
stock:ORLY, r

stock:ORLY, row: 1180
stock:ORLY, row: 1181
stock:ORLY, row: 1182
stock:ORLY, row: 1183
stock:ORLY, row: 1184
stock:ORLY, row: 1185
stock:ORLY, row: 1186
stock:ORLY, row: 1187
stock:ORLY, row: 1188
stock:ORLY, row: 1189
stock:ORLY, row: 1190
stock:ORLY, row: 1191
stock:ORLY, row: 1192
stock:ORLY, row: 1193
stock:ORLY, row: 1194
stock:ORLY, row: 1195
stock:ORLY, row: 1196
stock:ORLY, row: 1197
stock:ORLY, row: 1198
stock:ORLY, row: 1199
stock:ORLY, row: 1200
stock:ORLY, row: 1201
stock:ORLY, row: 1202
stock:ORLY, row: 1203
stock:ORLY, row: 1204
stock:ORLY, row: 1205
stock:ORLY, row: 1206
stock:ORLY, row: 1207
stock:ORLY, row: 1208
stock:ORLY, row: 1209
stock:ORLY, row: 1210
stock:ORLY, row: 1211
stock:ORLY, row: 1212
stock:ORLY, row: 1213
stock:ORLY, row: 1214
stock:ORLY, row: 1215
stock:ORLY, row: 1216
stock:ORLY, row: 1217
stock:ORLY, row: 1218
stock:ORLY, row: 1219
stock:ORLY, row: 1220
stock:ORLY, row: 1221
stock:ORLY, row: 1222
stock:ORLY, row: 1223
stock:ORLY, row: 1224
stock:ORLY

stock:ORLY, row: 1726
stock:ORLY, row: 1727
stock:ORLY, row: 1728
stock:ORLY, row: 1729
stock:ORLY, row: 1730
stock:ORLY, row: 1731
stock:ORLY, row: 1732
stock:ORLY, row: 1733
stock:ORLY, row: 1734
stock:ORLY, row: 1735
stock:ORLY, row: 1736
stock:ORLY, row: 1737
stock:ORLY, row: 1738
stock:ORLY, row: 1739
stock:ORLY, row: 1740
stock:ORLY, row: 1741
stock:ORLY, row: 1742
stock:ORLY, row: 1743
stock:ORLY, row: 1744
stock:ORLY, row: 1745
stock:ORLY, row: 1746
stock:ORLY, row: 1747
stock:ORLY, row: 1748
stock:ORLY, row: 1749
stock:ORLY, row: 1750
stock:ORLY, row: 1751
stock:ORLY, row: 1752
stock:ORLY, row: 1753
stock:ORLY, row: 1754
stock:ORLY, row: 1755
stock:ORLY, row: 1756
stock:ORLY, row: 1757
stock:ORLY, row: 1758
stock:ORLY, row: 1759
stock:ORLY, row: 1760
stock:ORLY, row: 1761
stock:ORLY, row: 1762
stock:ORLY, row: 1763
stock:ORLY, row: 1764
stock:ORLY, row: 1765
stock:ORLY, row: 1766
stock:ORLY, row: 1767
stock:ORLY, row: 1768
stock:ORLY, row: 1769
stock:ORLY, row: 1770
stock:ORLY

stock:ORLY, row: 2501
stock:ORLY, row: 2502
stock:ORLY, row: 2503
stock:ORLY, row: 2504
stock:ORLY, row: 2505
stock:ORLY, row: 2506
stock:ORLY, row: 2507
stock:ORLY, row: 2508
stock:ORLY, row: 2509
stock:ORLY, row: 2510
stock:ORLY, row: 2511
stock:ORLY, row: 2512
stock:ORLY, row: 2513
stock:ORLY, row: 2514
stock:ORLY, row: 2515
stock:ORLY, row: 2516
stock:ORLY, row: 2517
stock:ORLY, row: 2518
stock:ORLY, row: 2519
stock:ORLY, row: 2520
stock:ORLY, row: 2521
stock:ORLY, row: 2522
stock:ORLY, row: 2523
stock:ORLY, row: 2524
stock:ORLY, row: 2525
stock:ORLY, row: 2526
stock:ORLY, row: 2527
stock:ORLY, row: 2528
stock:ORLY, row: 2529
stock:ORLY, row: 2530
stock:ORLY, row: 2531
stock:ORLY, row: 2532
stock:ORLY, row: 2533
stock:ORLY, row: 2534
stock:ORLY, row: 2535
stock:ORLY, row: 2536
stock:ORLY, row: 2537
stock:ORLY, row: 2538
stock:ORLY, row: 2539
stock:ORLY, row: 2540
stock:ORLY, row: 2541
stock:ORLY, row: 2542
stock:ORLY, row: 2543
stock:ORLY, row: 2544
stock:ORLY, row: 2545
stock:ORLY

stock:BSX, row: 120
stock:BSX, row: 121
stock:BSX, row: 122
stock:BSX, row: 123
stock:BSX, row: 124
stock:BSX, row: 125
stock:BSX, row: 126
stock:BSX, row: 127
stock:BSX, row: 128
stock:BSX, row: 129
stock:BSX, row: 130
stock:BSX, row: 131
stock:BSX, row: 132
stock:BSX, row: 133
stock:BSX, row: 134
stock:BSX, row: 135
stock:BSX, row: 136
stock:BSX, row: 137
stock:BSX, row: 138
stock:BSX, row: 139
stock:BSX, row: 140
stock:BSX, row: 141
stock:BSX, row: 142
stock:BSX, row: 143
stock:BSX, row: 144
stock:BSX, row: 145
stock:BSX, row: 146
stock:BSX, row: 147
stock:BSX, row: 148
stock:BSX, row: 149
stock:BSX, row: 150
stock:BSX, row: 151
stock:BSX, row: 152
stock:BSX, row: 153
stock:BSX, row: 154
stock:BSX, row: 155
stock:BSX, row: 156
stock:BSX, row: 157
stock:BSX, row: 158
stock:BSX, row: 159
stock:BSX, row: 160
stock:BSX, row: 161
stock:BSX, row: 162
stock:BSX, row: 163
stock:BSX, row: 164
stock:BSX, row: 165
stock:BSX, row: 166
stock:BSX, row: 167
stock:BSX, row: 168
stock:BSX, row: 169


stock:BSX, row: 968
stock:BSX, row: 969
stock:BSX, row: 970
stock:BSX, row: 971
stock:BSX, row: 972
stock:BSX, row: 973
stock:BSX, row: 974
stock:BSX, row: 975
stock:BSX, row: 976
stock:BSX, row: 977
stock:BSX, row: 978
stock:BSX, row: 979
stock:BSX, row: 980
stock:BSX, row: 981
stock:BSX, row: 982
stock:BSX, row: 983
stock:BSX, row: 984
stock:BSX, row: 985
stock:BSX, row: 986
stock:BSX, row: 987
stock:BSX, row: 988
stock:BSX, row: 989
stock:BSX, row: 990
stock:BSX, row: 991
stock:BSX, row: 992
stock:BSX, row: 993
stock:BSX, row: 994
stock:BSX, row: 995
stock:BSX, row: 996
stock:BSX, row: 997
stock:BSX, row: 998
stock:BSX, row: 999
stock:BSX, row: 1000
stock:BSX, row: 1001
stock:BSX, row: 1002
stock:BSX, row: 1003
stock:BSX, row: 1004
stock:BSX, row: 1005
stock:BSX, row: 1006
stock:BSX, row: 1007
stock:BSX, row: 1008
stock:BSX, row: 1009
stock:BSX, row: 1010
stock:BSX, row: 1011
stock:BSX, row: 1012
stock:BSX, row: 1013
stock:BSX, row: 1014
stock:BSX, row: 1015
stock:BSX, row: 1016
sto

stock:BSX, row: 1890
stock:BSX, row: 1891
stock:BSX, row: 1892
stock:BSX, row: 1893
stock:BSX, row: 1894
stock:BSX, row: 1895
stock:BSX, row: 1896
stock:BSX, row: 1897
stock:BSX, row: 1898
stock:BSX, row: 1899
stock:BSX, row: 1900
stock:BSX, row: 1901
stock:BSX, row: 1902
stock:BSX, row: 1903
stock:BSX, row: 1904
stock:BSX, row: 1905
stock:BSX, row: 1906
stock:BSX, row: 1907
stock:BSX, row: 1908
stock:BSX, row: 1909
stock:BSX, row: 1910
stock:BSX, row: 1911
stock:BSX, row: 1912
stock:BSX, row: 1913
stock:BSX, row: 1914
stock:BSX, row: 1915
stock:BSX, row: 1916
stock:BSX, row: 1917
stock:BSX, row: 1918
stock:BSX, row: 1919
stock:BSX, row: 1920
stock:BSX, row: 1921
stock:BSX, row: 1922
stock:BSX, row: 1923
stock:BSX, row: 1924
stock:BSX, row: 1925
stock:BSX, row: 1926
stock:BSX, row: 1927
stock:BSX, row: 1928
stock:BSX, row: 1929
stock:BSX, row: 1930
stock:BSX, row: 1931
stock:BSX, row: 1932
stock:BSX, row: 1933
stock:BSX, row: 1934
stock:BSX, row: 1935
stock:BSX, row: 1936
stock:BSX, ro

stock:BSX, row: 2963
stock:BSX, row: 2964
stock:BSX, row: 2965
stock:BSX, row: 2966
stock:BSX, row: 2967
stock:BSX, row: 2968
stock:BSX, row: 2969
stock:BSX, row: 2970
stock:BSX, row: 2971
stock:BSX, row: 2972
stock:BSX, row: 2973
stock:BSX, row: 2974
stock:BSX, row: 2975
stock:BSX, row: 2976
stock:USB, row: 0
stock:USB, row: 1
stock:USB, row: 2
stock:USB, row: 3
stock:USB, row: 4
stock:USB, row: 5
stock:USB, row: 6
stock:USB, row: 7
stock:USB, row: 8
stock:USB, row: 9
stock:USB, row: 10
stock:USB, row: 11
stock:USB, row: 12
stock:USB, row: 13
stock:USB, row: 14
stock:USB, row: 15
stock:USB, row: 16
stock:USB, row: 17
stock:USB, row: 18
stock:USB, row: 19
stock:USB, row: 20
stock:USB, row: 21
stock:USB, row: 22
stock:USB, row: 23
stock:USB, row: 24
stock:USB, row: 25
stock:USB, row: 26
stock:USB, row: 27
stock:USB, row: 28
stock:USB, row: 29
stock:USB, row: 30
stock:USB, row: 31
stock:USB, row: 32
stock:USB, row: 33
stock:USB, row: 34
stock:USB, row: 35
stock:USB, row: 36
stock:USB, ro

stock:USB, row: 925
stock:USB, row: 926
stock:USB, row: 927
stock:USB, row: 928
stock:USB, row: 929
stock:USB, row: 930
stock:USB, row: 931
stock:USB, row: 932
stock:USB, row: 933
stock:USB, row: 934
stock:USB, row: 935
stock:USB, row: 936
stock:USB, row: 937
stock:USB, row: 938
stock:USB, row: 939
stock:USB, row: 940
stock:USB, row: 941
stock:USB, row: 942
stock:USB, row: 943
stock:USB, row: 944
stock:USB, row: 945
stock:USB, row: 946
stock:USB, row: 947
stock:USB, row: 948
stock:USB, row: 949
stock:USB, row: 950
stock:USB, row: 951
stock:USB, row: 952
stock:USB, row: 953
stock:USB, row: 954
stock:USB, row: 955
stock:USB, row: 956
stock:USB, row: 957
stock:USB, row: 958
stock:USB, row: 959
stock:USB, row: 960
stock:USB, row: 961
stock:USB, row: 962
stock:USB, row: 963
stock:USB, row: 964
stock:USB, row: 965
stock:USB, row: 966
stock:USB, row: 967
stock:USB, row: 968
stock:USB, row: 969
stock:USB, row: 970
stock:USB, row: 971
stock:USB, row: 972
stock:USB, row: 973
stock:USB, row: 974


stock:USB, row: 1388
stock:USB, row: 1389
stock:USB, row: 1390
stock:USB, row: 1391
stock:USB, row: 1392
stock:USB, row: 1393
stock:USB, row: 1394
stock:USB, row: 1395
stock:USB, row: 1396
stock:USB, row: 1397
stock:USB, row: 1398
stock:USB, row: 1399
stock:USB, row: 1400
stock:USB, row: 1401
stock:USB, row: 1402
stock:USB, row: 1403
stock:USB, row: 1404
stock:USB, row: 1405
stock:USB, row: 1406
stock:USB, row: 1407
stock:USB, row: 1408
stock:USB, row: 1409
stock:USB, row: 1410
stock:USB, row: 1411
stock:USB, row: 1412
stock:USB, row: 1413
stock:USB, row: 1414
stock:USB, row: 1415
stock:USB, row: 1416
stock:USB, row: 1417
stock:USB, row: 1418
stock:USB, row: 1419
stock:USB, row: 1420
stock:USB, row: 1421
stock:USB, row: 1422
stock:USB, row: 1423
stock:USB, row: 1424
stock:USB, row: 1425
stock:USB, row: 1426
stock:USB, row: 1427
stock:USB, row: 1428
stock:USB, row: 1429
stock:USB, row: 1430
stock:USB, row: 1431
stock:USB, row: 1432
stock:USB, row: 1433
stock:USB, row: 1434
stock:USB, ro

stock:USB, row: 2317
stock:USB, row: 2318
stock:USB, row: 2319
stock:USB, row: 2320
stock:USB, row: 2321
stock:USB, row: 2322
stock:USB, row: 2323
stock:USB, row: 2324
stock:USB, row: 2325
stock:USB, row: 2326
stock:USB, row: 2327
stock:USB, row: 2328
stock:USB, row: 2329
stock:USB, row: 2330
stock:USB, row: 2331
stock:USB, row: 2332
stock:USB, row: 2333
stock:USB, row: 2334
stock:USB, row: 2335
stock:USB, row: 2336
stock:USB, row: 2337
stock:USB, row: 2338
stock:USB, row: 2339
stock:USB, row: 2340
stock:USB, row: 2341
stock:USB, row: 2342
stock:USB, row: 2343
stock:USB, row: 2344
stock:USB, row: 2345
stock:USB, row: 2346
stock:USB, row: 2347
stock:USB, row: 2348
stock:USB, row: 2349
stock:USB, row: 2350
stock:USB, row: 2351
stock:USB, row: 2352
stock:USB, row: 2353
stock:USB, row: 2354
stock:USB, row: 2355
stock:USB, row: 2356
stock:USB, row: 2357
stock:USB, row: 2358
stock:USB, row: 2359
stock:USB, row: 2360
stock:USB, row: 2361
stock:USB, row: 2362
stock:USB, row: 2363
stock:USB, ro

stock:ETR, row: 111
stock:ETR, row: 112
stock:ETR, row: 113
stock:ETR, row: 114
stock:ETR, row: 115
stock:ETR, row: 116
stock:ETR, row: 117
stock:ETR, row: 118
stock:ETR, row: 119
stock:ETR, row: 120
stock:ETR, row: 121
stock:ETR, row: 122
stock:ETR, row: 123
stock:ETR, row: 124
stock:ETR, row: 125
stock:ETR, row: 126
stock:ETR, row: 127
stock:ETR, row: 128
stock:ETR, row: 129
stock:ETR, row: 130
stock:ETR, row: 131
stock:ETR, row: 132
stock:ETR, row: 133
stock:ETR, row: 134
stock:ETR, row: 135
stock:ETR, row: 136
stock:ETR, row: 137
stock:ETR, row: 138
stock:ETR, row: 139
stock:ETR, row: 140
stock:ETR, row: 141
stock:ETR, row: 142
stock:ETR, row: 143
stock:ETR, row: 144
stock:ETR, row: 145
stock:ETR, row: 146
stock:ETR, row: 147
stock:ETR, row: 148
stock:ETR, row: 149
stock:ETR, row: 150
stock:ETR, row: 151
stock:ETR, row: 152
stock:ETR, row: 153
stock:ETR, row: 154
stock:ETR, row: 155
stock:ETR, row: 156
stock:ETR, row: 157
stock:ETR, row: 158
stock:ETR, row: 159
stock:ETR, row: 160


stock:ETR, row: 608
stock:ETR, row: 609
stock:ETR, row: 610
stock:ETR, row: 611
stock:ETR, row: 612
stock:ETR, row: 613
stock:ETR, row: 614
stock:ETR, row: 615
stock:ETR, row: 616
stock:ETR, row: 617
stock:ETR, row: 618
stock:ETR, row: 619
stock:ETR, row: 620
stock:ETR, row: 621
stock:ETR, row: 622
stock:ETR, row: 623
stock:ETR, row: 624
stock:ETR, row: 625
stock:ETR, row: 626
stock:ETR, row: 627
stock:ETR, row: 628
stock:ETR, row: 629
stock:ETR, row: 630
stock:ETR, row: 631
stock:ETR, row: 632
stock:ETR, row: 633
stock:ETR, row: 634
stock:ETR, row: 635
stock:ETR, row: 636
stock:ETR, row: 637
stock:ETR, row: 638
stock:ETR, row: 639
stock:ETR, row: 640
stock:ETR, row: 641
stock:ETR, row: 642
stock:ETR, row: 643
stock:ETR, row: 644
stock:ETR, row: 645
stock:ETR, row: 646
stock:ETR, row: 647
stock:ETR, row: 648
stock:ETR, row: 649
stock:ETR, row: 650
stock:ETR, row: 651
stock:ETR, row: 652
stock:ETR, row: 653
stock:ETR, row: 654
stock:ETR, row: 655
stock:ETR, row: 656
stock:ETR, row: 657


stock:ETR, row: 1825
stock:ETR, row: 1826
stock:ETR, row: 1827
stock:ETR, row: 1828
stock:ETR, row: 1829
stock:ETR, row: 1830
stock:ETR, row: 1831
stock:ETR, row: 1832
stock:ETR, row: 1833
stock:ETR, row: 1834
stock:ETR, row: 1835
stock:ETR, row: 1836
stock:ETR, row: 1837
stock:ETR, row: 1838
stock:ETR, row: 1839
stock:ETR, row: 1840
stock:ETR, row: 1841
stock:ETR, row: 1842
stock:ETR, row: 1843
stock:ETR, row: 1844
stock:ETR, row: 1845
stock:ETR, row: 1846
stock:ETR, row: 1847
stock:ETR, row: 1848
stock:ETR, row: 1849
stock:ETR, row: 1850
stock:ETR, row: 1851
stock:ETR, row: 1852
stock:ETR, row: 1853
stock:ETR, row: 1854
stock:ETR, row: 1855
stock:ETR, row: 1856
stock:ETR, row: 1857
stock:ETR, row: 1858
stock:ETR, row: 1859
stock:ETR, row: 1860
stock:ETR, row: 1861
stock:ETR, row: 1862
stock:ETR, row: 1863
stock:ETR, row: 1864
stock:ETR, row: 1865
stock:ETR, row: 1866
stock:ETR, row: 1867
stock:ETR, row: 1868
stock:ETR, row: 1869
stock:ETR, row: 1870
stock:ETR, row: 1871
stock:ETR, ro

stock:ETR, row: 2569
stock:ETR, row: 2570
stock:ETR, row: 2571
stock:ETR, row: 2572
stock:ETR, row: 2573
stock:ETR, row: 2574
stock:ETR, row: 2575
stock:ETR, row: 2576
stock:ETR, row: 2577
stock:ETR, row: 2578
stock:ETR, row: 2579
stock:ETR, row: 2580
stock:ETR, row: 2581
stock:ETR, row: 2582
stock:ETR, row: 2583
stock:ETR, row: 2584
stock:ETR, row: 2585
stock:ETR, row: 2586
stock:ETR, row: 2587
stock:ETR, row: 2588
stock:ETR, row: 2589
stock:ETR, row: 2590
stock:ETR, row: 2591
stock:ETR, row: 2592
stock:ETR, row: 2593
stock:ETR, row: 2594
stock:ETR, row: 2595
stock:ETR, row: 2596
stock:ETR, row: 2597
stock:ETR, row: 2598
stock:ETR, row: 2599
stock:ETR, row: 2600
stock:ETR, row: 2601
stock:ETR, row: 2602
stock:ETR, row: 2603
stock:ETR, row: 2604
stock:ETR, row: 2605
stock:ETR, row: 2606
stock:ETR, row: 2607
stock:ETR, row: 2608
stock:ETR, row: 2609
stock:ETR, row: 2610
stock:ETR, row: 2611
stock:ETR, row: 2612
stock:ETR, row: 2613
stock:ETR, row: 2614
stock:ETR, row: 2615
stock:ETR, ro

stock:AZO, row: 555
stock:AZO, row: 556
stock:AZO, row: 557
stock:AZO, row: 558
stock:AZO, row: 559
stock:AZO, row: 560
stock:AZO, row: 561
stock:AZO, row: 562
stock:AZO, row: 563
stock:AZO, row: 564
stock:AZO, row: 565
stock:AZO, row: 566
stock:AZO, row: 567
stock:AZO, row: 568
stock:AZO, row: 569
stock:AZO, row: 570
stock:AZO, row: 571
stock:AZO, row: 572
stock:AZO, row: 573
stock:AZO, row: 574
stock:AZO, row: 575
stock:AZO, row: 576
stock:AZO, row: 577
stock:AZO, row: 578
stock:AZO, row: 579
stock:AZO, row: 580
stock:AZO, row: 581
stock:AZO, row: 582
stock:AZO, row: 583
stock:AZO, row: 584
stock:AZO, row: 585
stock:AZO, row: 586
stock:AZO, row: 587
stock:AZO, row: 588
stock:AZO, row: 589
stock:AZO, row: 590
stock:AZO, row: 591
stock:AZO, row: 592
stock:AZO, row: 593
stock:AZO, row: 594
stock:AZO, row: 595
stock:AZO, row: 596
stock:AZO, row: 597
stock:AZO, row: 598
stock:AZO, row: 599
stock:AZO, row: 600
stock:AZO, row: 601
stock:AZO, row: 602
stock:AZO, row: 603
stock:AZO, row: 604


stock:AZO, row: 1071
stock:AZO, row: 1072
stock:AZO, row: 1073
stock:AZO, row: 1074
stock:AZO, row: 1075
stock:AZO, row: 1076
stock:AZO, row: 1077
stock:AZO, row: 1078
stock:AZO, row: 1079
stock:AZO, row: 1080
stock:AZO, row: 1081
stock:AZO, row: 1082
stock:AZO, row: 1083
stock:AZO, row: 1084
stock:AZO, row: 1085
stock:AZO, row: 1086
stock:AZO, row: 1087
stock:AZO, row: 1088
stock:AZO, row: 1089
stock:AZO, row: 1090
stock:AZO, row: 1091
stock:AZO, row: 1092
stock:AZO, row: 1093
stock:AZO, row: 1094
stock:AZO, row: 1095
stock:AZO, row: 1096
stock:AZO, row: 1097
stock:AZO, row: 1098
stock:AZO, row: 1099
stock:AZO, row: 1100
stock:AZO, row: 1101
stock:AZO, row: 1102
stock:AZO, row: 1103
stock:AZO, row: 1104
stock:AZO, row: 1105
stock:AZO, row: 1106
stock:AZO, row: 1107
stock:AZO, row: 1108
stock:AZO, row: 1109
stock:AZO, row: 1110
stock:AZO, row: 1111
stock:AZO, row: 1112
stock:AZO, row: 1113
stock:AZO, row: 1114
stock:AZO, row: 1115
stock:AZO, row: 1116
stock:AZO, row: 1117
stock:AZO, ro

stock:AZO, row: 1976
stock:AZO, row: 1977
stock:AZO, row: 1978
stock:AZO, row: 1979
stock:AZO, row: 1980
stock:AZO, row: 1981
stock:AZO, row: 1982
stock:AZO, row: 1983
stock:AZO, row: 1984
stock:AZO, row: 1985
stock:AZO, row: 1986
stock:AZO, row: 1987
stock:AZO, row: 1988
stock:AZO, row: 1989
stock:AZO, row: 1990
stock:AZO, row: 1991
stock:AZO, row: 1992
stock:AZO, row: 1993
stock:AZO, row: 1994
stock:AZO, row: 1995
stock:AZO, row: 1996
stock:AZO, row: 1997
stock:AZO, row: 1998
stock:AZO, row: 1999
stock:AZO, row: 2000
stock:AZO, row: 2001
stock:AZO, row: 2002
stock:AZO, row: 2003
stock:AZO, row: 2004
stock:AZO, row: 2005
stock:AZO, row: 2006
stock:AZO, row: 2007
stock:AZO, row: 2008
stock:AZO, row: 2009
stock:AZO, row: 2010
stock:AZO, row: 2011
stock:AZO, row: 2012
stock:AZO, row: 2013
stock:AZO, row: 2014
stock:AZO, row: 2015
stock:AZO, row: 2016
stock:AZO, row: 2017
stock:AZO, row: 2018
stock:AZO, row: 2019
stock:AZO, row: 2020
stock:AZO, row: 2021
stock:AZO, row: 2022
stock:AZO, ro

stock:AZO, row: 2691
stock:AZO, row: 2692
stock:AZO, row: 2693
stock:AZO, row: 2694
stock:AZO, row: 2695
stock:AZO, row: 2696
stock:AZO, row: 2697
stock:AZO, row: 2698
stock:AZO, row: 2699
stock:AZO, row: 2700
stock:AZO, row: 2701
stock:AZO, row: 2702
stock:AZO, row: 2703
stock:AZO, row: 2704
stock:AZO, row: 2705
stock:AZO, row: 2706
stock:AZO, row: 2707
stock:AZO, row: 2708
stock:AZO, row: 2709
stock:AZO, row: 2710
stock:AZO, row: 2711
stock:AZO, row: 2712
stock:AZO, row: 2713
stock:AZO, row: 2714
stock:AZO, row: 2715
stock:AZO, row: 2716
stock:AZO, row: 2717
stock:AZO, row: 2718
stock:AZO, row: 2719
stock:AZO, row: 2720
stock:AZO, row: 2721
stock:AZO, row: 2722
stock:AZO, row: 2723
stock:AZO, row: 2724
stock:AZO, row: 2725
stock:AZO, row: 2726
stock:AZO, row: 2727
stock:AZO, row: 2728
stock:AZO, row: 2729
stock:AZO, row: 2730
stock:AZO, row: 2731
stock:AZO, row: 2732
stock:AZO, row: 2733
stock:AZO, row: 2734
stock:AZO, row: 2735
stock:AZO, row: 2736
stock:AZO, row: 2737
stock:AZO, ro

stock:PVH, row: 397
stock:PVH, row: 398
stock:PVH, row: 399
stock:PVH, row: 400
stock:PVH, row: 401
stock:PVH, row: 402
stock:PVH, row: 403
stock:PVH, row: 404
stock:PVH, row: 405
stock:PVH, row: 406
stock:PVH, row: 407
stock:PVH, row: 408
stock:PVH, row: 409
stock:PVH, row: 410
stock:PVH, row: 411
stock:PVH, row: 412
stock:PVH, row: 413
stock:PVH, row: 414
stock:PVH, row: 415
stock:PVH, row: 416
stock:PVH, row: 417
stock:PVH, row: 418
stock:PVH, row: 419
stock:PVH, row: 420
stock:PVH, row: 421
stock:PVH, row: 422
stock:PVH, row: 423
stock:PVH, row: 424
stock:PVH, row: 425
stock:PVH, row: 426
stock:PVH, row: 427
stock:PVH, row: 428
stock:PVH, row: 429
stock:PVH, row: 430
stock:PVH, row: 431
stock:PVH, row: 432
stock:PVH, row: 433
stock:PVH, row: 434
stock:PVH, row: 435
stock:PVH, row: 436
stock:PVH, row: 437
stock:PVH, row: 438
stock:PVH, row: 439
stock:PVH, row: 440
stock:PVH, row: 441
stock:PVH, row: 442
stock:PVH, row: 443
stock:PVH, row: 444
stock:PVH, row: 445
stock:PVH, row: 446


stock:PVH, row: 933
stock:PVH, row: 934
stock:PVH, row: 935
stock:PVH, row: 936
stock:PVH, row: 937
stock:PVH, row: 938
stock:PVH, row: 939
stock:PVH, row: 940
stock:PVH, row: 941
stock:PVH, row: 942
stock:PVH, row: 943
stock:PVH, row: 944
stock:PVH, row: 945
stock:PVH, row: 946
stock:PVH, row: 947
stock:PVH, row: 948
stock:PVH, row: 949
stock:PVH, row: 950
stock:PVH, row: 951
stock:PVH, row: 952
stock:PVH, row: 953
stock:PVH, row: 954
stock:PVH, row: 955
stock:PVH, row: 956
stock:PVH, row: 957
stock:PVH, row: 958
stock:PVH, row: 959
stock:PVH, row: 960
stock:PVH, row: 961
stock:PVH, row: 962
stock:PVH, row: 963
stock:PVH, row: 964
stock:PVH, row: 965
stock:PVH, row: 966
stock:PVH, row: 967
stock:PVH, row: 968
stock:PVH, row: 969
stock:PVH, row: 970
stock:PVH, row: 971
stock:PVH, row: 972
stock:PVH, row: 973
stock:PVH, row: 974
stock:PVH, row: 975
stock:PVH, row: 976
stock:PVH, row: 977
stock:PVH, row: 978
stock:PVH, row: 979
stock:PVH, row: 980
stock:PVH, row: 981
stock:PVH, row: 982


stock:PVH, row: 1724
stock:PVH, row: 1725
stock:PVH, row: 1726
stock:PVH, row: 1727
stock:PVH, row: 1728
stock:PVH, row: 1729
stock:PVH, row: 1730
stock:PVH, row: 1731
stock:PVH, row: 1732
stock:PVH, row: 1733
stock:PVH, row: 1734
stock:PVH, row: 1735
stock:PVH, row: 1736
stock:PVH, row: 1737
stock:PVH, row: 1738
stock:PVH, row: 1739
stock:PVH, row: 1740
stock:PVH, row: 1741
stock:PVH, row: 1742
stock:PVH, row: 1743
stock:PVH, row: 1744
stock:PVH, row: 1745
stock:PVH, row: 1746
stock:PVH, row: 1747
stock:PVH, row: 1748
stock:PVH, row: 1749
stock:PVH, row: 1750
stock:PVH, row: 1751
stock:PVH, row: 1752
stock:PVH, row: 1753
stock:PVH, row: 1754
stock:PVH, row: 1755
stock:PVH, row: 1756
stock:PVH, row: 1757
stock:PVH, row: 1758
stock:PVH, row: 1759
stock:PVH, row: 1760
stock:PVH, row: 1761
stock:PVH, row: 1762
stock:PVH, row: 1763
stock:PVH, row: 1764
stock:PVH, row: 1765
stock:PVH, row: 1766
stock:PVH, row: 1767
stock:PVH, row: 1768
stock:PVH, row: 1769
stock:PVH, row: 1770
stock:PVH, ro

stock:PVH, row: 2721
stock:PVH, row: 2722
stock:PVH, row: 2723
stock:PVH, row: 2724
stock:PVH, row: 2725
stock:PVH, row: 2726
stock:PVH, row: 2727
stock:PVH, row: 2728
stock:PVH, row: 2729
stock:PVH, row: 2730
stock:PVH, row: 2731
stock:PVH, row: 2732
stock:PVH, row: 2733
stock:PVH, row: 2734
stock:PVH, row: 2735
stock:PVH, row: 2736
stock:PVH, row: 2737
stock:PVH, row: 2738
stock:PVH, row: 2739
stock:PVH, row: 2740
stock:PVH, row: 2741
stock:PVH, row: 2742
stock:PVH, row: 2743
stock:PVH, row: 2744
stock:PVH, row: 2745
stock:PVH, row: 2746
stock:PVH, row: 2747
stock:PVH, row: 2748
stock:PVH, row: 2749
stock:PVH, row: 2750
stock:PVH, row: 2751
stock:PVH, row: 2752
stock:PVH, row: 2753
stock:PVH, row: 2754
stock:PVH, row: 2755
stock:PVH, row: 2756
stock:PVH, row: 2757
stock:PVH, row: 2758
stock:PVH, row: 2759
stock:PVH, row: 2760
stock:PVH, row: 2761
stock:PVH, row: 2762
stock:PVH, row: 2763
stock:PVH, row: 2764
stock:PVH, row: 2765
stock:PVH, row: 2766
stock:PVH, row: 2767
stock:PVH, ro

stock:BIIB, row: 760
stock:BIIB, row: 761
stock:BIIB, row: 762
stock:BIIB, row: 763
stock:BIIB, row: 764
stock:BIIB, row: 765
stock:BIIB, row: 766
stock:BIIB, row: 767
stock:BIIB, row: 768
stock:BIIB, row: 769
stock:BIIB, row: 770
stock:BIIB, row: 771
stock:BIIB, row: 772
stock:BIIB, row: 773
stock:BIIB, row: 774
stock:BIIB, row: 775
stock:BIIB, row: 776
stock:BIIB, row: 777
stock:BIIB, row: 778
stock:BIIB, row: 779
stock:BIIB, row: 780
stock:BIIB, row: 781
stock:BIIB, row: 782
stock:BIIB, row: 783
stock:BIIB, row: 784
stock:BIIB, row: 785
stock:BIIB, row: 786
stock:BIIB, row: 787
stock:BIIB, row: 788
stock:BIIB, row: 789
stock:BIIB, row: 790
stock:BIIB, row: 791
stock:BIIB, row: 792
stock:BIIB, row: 793
stock:BIIB, row: 794
stock:BIIB, row: 795
stock:BIIB, row: 796
stock:BIIB, row: 797
stock:BIIB, row: 798
stock:BIIB, row: 799
stock:BIIB, row: 800
stock:BIIB, row: 801
stock:BIIB, row: 802
stock:BIIB, row: 803
stock:BIIB, row: 804
stock:BIIB, row: 805
stock:BIIB, row: 806
stock:BIIB, r

stock:BIIB, row: 2109
stock:BIIB, row: 2110
stock:BIIB, row: 2111
stock:BIIB, row: 2112
stock:BIIB, row: 2113
stock:BIIB, row: 2114
stock:BIIB, row: 2115
stock:BIIB, row: 2116
stock:BIIB, row: 2117
stock:BIIB, row: 2118
stock:BIIB, row: 2119
stock:BIIB, row: 2120
stock:BIIB, row: 2121
stock:BIIB, row: 2122
stock:BIIB, row: 2123
stock:BIIB, row: 2124
stock:BIIB, row: 2125
stock:BIIB, row: 2126
stock:BIIB, row: 2127
stock:BIIB, row: 2128
stock:BIIB, row: 2129
stock:BIIB, row: 2130
stock:BIIB, row: 2131
stock:BIIB, row: 2132
stock:BIIB, row: 2133
stock:BIIB, row: 2134
stock:BIIB, row: 2135
stock:BIIB, row: 2136
stock:BIIB, row: 2137
stock:BIIB, row: 2138
stock:BIIB, row: 2139
stock:BIIB, row: 2140
stock:BIIB, row: 2141
stock:BIIB, row: 2142
stock:BIIB, row: 2143
stock:BIIB, row: 2144
stock:BIIB, row: 2145
stock:BIIB, row: 2146
stock:BIIB, row: 2147
stock:BIIB, row: 2148
stock:BIIB, row: 2149
stock:BIIB, row: 2150
stock:BIIB, row: 2151
stock:BIIB, row: 2152
stock:BIIB, row: 2153
stock:BIIB

stock:ETN, row: 167
stock:ETN, row: 168
stock:ETN, row: 169
stock:ETN, row: 170
stock:ETN, row: 171
stock:ETN, row: 172
stock:ETN, row: 173
stock:ETN, row: 174
stock:ETN, row: 175
stock:ETN, row: 176
stock:ETN, row: 177
stock:ETN, row: 178
stock:ETN, row: 179
stock:ETN, row: 180
stock:ETN, row: 181
stock:ETN, row: 182
stock:ETN, row: 183
stock:ETN, row: 184
stock:ETN, row: 185
stock:ETN, row: 186
stock:ETN, row: 187
stock:ETN, row: 188
stock:ETN, row: 189
stock:ETN, row: 190
stock:ETN, row: 191
stock:ETN, row: 192
stock:ETN, row: 193
stock:ETN, row: 194
stock:ETN, row: 195
stock:ETN, row: 196
stock:ETN, row: 197
stock:ETN, row: 198
stock:ETN, row: 199
stock:ETN, row: 200
stock:ETN, row: 201
stock:ETN, row: 202
stock:ETN, row: 203
stock:ETN, row: 204
stock:ETN, row: 205
stock:ETN, row: 206
stock:ETN, row: 207
stock:ETN, row: 208
stock:ETN, row: 209
stock:ETN, row: 210
stock:ETN, row: 211
stock:ETN, row: 212
stock:ETN, row: 213
stock:ETN, row: 214
stock:ETN, row: 215
stock:ETN, row: 216


stock:ETN, row: 1206
stock:ETN, row: 1207
stock:ETN, row: 1208
stock:ETN, row: 1209
stock:ETN, row: 1210
stock:ETN, row: 1211
stock:ETN, row: 1212
stock:ETN, row: 1213
stock:ETN, row: 1214
stock:ETN, row: 1215
stock:ETN, row: 1216
stock:ETN, row: 1217
stock:ETN, row: 1218
stock:ETN, row: 1219
stock:ETN, row: 1220
stock:ETN, row: 1221
stock:ETN, row: 1222
stock:ETN, row: 1223
stock:ETN, row: 1224
stock:ETN, row: 1225
stock:ETN, row: 1226
stock:ETN, row: 1227
stock:ETN, row: 1228
stock:ETN, row: 1229
stock:ETN, row: 1230
stock:ETN, row: 1231
stock:ETN, row: 1232
stock:ETN, row: 1233
stock:ETN, row: 1234
stock:ETN, row: 1235
stock:ETN, row: 1236
stock:ETN, row: 1237
stock:ETN, row: 1238
stock:ETN, row: 1239
stock:ETN, row: 1240
stock:ETN, row: 1241
stock:ETN, row: 1242
stock:ETN, row: 1243
stock:ETN, row: 1244
stock:ETN, row: 1245
stock:ETN, row: 1246
stock:ETN, row: 1247
stock:ETN, row: 1248
stock:ETN, row: 1249
stock:ETN, row: 1250
stock:ETN, row: 1251
stock:ETN, row: 1252
stock:ETN, ro

stock:ETN, row: 2049
stock:ETN, row: 2050
stock:ETN, row: 2051
stock:ETN, row: 2052
stock:ETN, row: 2053
stock:ETN, row: 2054
stock:ETN, row: 2055
stock:ETN, row: 2056
stock:ETN, row: 2057
stock:ETN, row: 2058
stock:ETN, row: 2059
stock:ETN, row: 2060
stock:ETN, row: 2061
stock:ETN, row: 2062
stock:ETN, row: 2063
stock:ETN, row: 2064
stock:ETN, row: 2065
stock:ETN, row: 2066
stock:ETN, row: 2067
stock:ETN, row: 2068
stock:ETN, row: 2069
stock:ETN, row: 2070
stock:ETN, row: 2071
stock:ETN, row: 2072
stock:ETN, row: 2073
stock:ETN, row: 2074
stock:ETN, row: 2075
stock:ETN, row: 2076
stock:ETN, row: 2077
stock:ETN, row: 2078
stock:ETN, row: 2079
stock:ETN, row: 2080
stock:ETN, row: 2081
stock:ETN, row: 2082
stock:ETN, row: 2083
stock:ETN, row: 2084
stock:ETN, row: 2085
stock:ETN, row: 2086
stock:ETN, row: 2087
stock:ETN, row: 2088
stock:ETN, row: 2089
stock:ETN, row: 2090
stock:ETN, row: 2091
stock:ETN, row: 2092
stock:ETN, row: 2093
stock:ETN, row: 2094
stock:ETN, row: 2095
stock:ETN, ro

stock:PFE, row: 85
stock:PFE, row: 86
stock:PFE, row: 87
stock:PFE, row: 88
stock:PFE, row: 89
stock:PFE, row: 90
stock:PFE, row: 91
stock:PFE, row: 92
stock:PFE, row: 93
stock:PFE, row: 94
stock:PFE, row: 95
stock:PFE, row: 96
stock:PFE, row: 97
stock:PFE, row: 98
stock:PFE, row: 99
stock:PFE, row: 100
stock:PFE, row: 101
stock:PFE, row: 102
stock:PFE, row: 103
stock:PFE, row: 104
stock:PFE, row: 105
stock:PFE, row: 106
stock:PFE, row: 107
stock:PFE, row: 108
stock:PFE, row: 109
stock:PFE, row: 110
stock:PFE, row: 111
stock:PFE, row: 112
stock:PFE, row: 113
stock:PFE, row: 114
stock:PFE, row: 115
stock:PFE, row: 116
stock:PFE, row: 117
stock:PFE, row: 118
stock:PFE, row: 119
stock:PFE, row: 120
stock:PFE, row: 121
stock:PFE, row: 122
stock:PFE, row: 123
stock:PFE, row: 124
stock:PFE, row: 125
stock:PFE, row: 126
stock:PFE, row: 127
stock:PFE, row: 128
stock:PFE, row: 129
stock:PFE, row: 130
stock:PFE, row: 131
stock:PFE, row: 132
stock:PFE, row: 133
stock:PFE, row: 134
stock:PFE, row:

stock:PFE, row: 571
stock:PFE, row: 572
stock:PFE, row: 573
stock:PFE, row: 574
stock:PFE, row: 575
stock:PFE, row: 576
stock:PFE, row: 577
stock:PFE, row: 578
stock:PFE, row: 579
stock:PFE, row: 580
stock:PFE, row: 581
stock:PFE, row: 582
stock:PFE, row: 583
stock:PFE, row: 584
stock:PFE, row: 585
stock:PFE, row: 586
stock:PFE, row: 587
stock:PFE, row: 588
stock:PFE, row: 589
stock:PFE, row: 590
stock:PFE, row: 591
stock:PFE, row: 592
stock:PFE, row: 593
stock:PFE, row: 594
stock:PFE, row: 595
stock:PFE, row: 596
stock:PFE, row: 597
stock:PFE, row: 598
stock:PFE, row: 599
stock:PFE, row: 600
stock:PFE, row: 601
stock:PFE, row: 602
stock:PFE, row: 603
stock:PFE, row: 604
stock:PFE, row: 605
stock:PFE, row: 606
stock:PFE, row: 607
stock:PFE, row: 608
stock:PFE, row: 609
stock:PFE, row: 610
stock:PFE, row: 611
stock:PFE, row: 612
stock:PFE, row: 613
stock:PFE, row: 614
stock:PFE, row: 615
stock:PFE, row: 616
stock:PFE, row: 617
stock:PFE, row: 618
stock:PFE, row: 619
stock:PFE, row: 620


stock:PFE, row: 1330
stock:PFE, row: 1331
stock:PFE, row: 1332
stock:PFE, row: 1333
stock:PFE, row: 1334
stock:PFE, row: 1335
stock:PFE, row: 1336
stock:PFE, row: 1337
stock:PFE, row: 1338
stock:PFE, row: 1339
stock:PFE, row: 1340
stock:PFE, row: 1341
stock:PFE, row: 1342
stock:PFE, row: 1343
stock:PFE, row: 1344
stock:PFE, row: 1345
stock:PFE, row: 1346
stock:PFE, row: 1347
stock:PFE, row: 1348
stock:PFE, row: 1349
stock:PFE, row: 1350
stock:PFE, row: 1351
stock:PFE, row: 1352
stock:PFE, row: 1353
stock:PFE, row: 1354
stock:PFE, row: 1355
stock:PFE, row: 1356
stock:PFE, row: 1357
stock:PFE, row: 1358
stock:PFE, row: 1359
stock:PFE, row: 1360
stock:PFE, row: 1361
stock:PFE, row: 1362
stock:PFE, row: 1363
stock:PFE, row: 1364
stock:PFE, row: 1365
stock:PFE, row: 1366
stock:PFE, row: 1367
stock:PFE, row: 1368
stock:PFE, row: 1369
stock:PFE, row: 1370
stock:PFE, row: 1371
stock:PFE, row: 1372
stock:PFE, row: 1373
stock:PFE, row: 1374
stock:PFE, row: 1375
stock:PFE, row: 1376
stock:PFE, ro

stock:PFE, row: 1866
stock:PFE, row: 1867
stock:PFE, row: 1868
stock:PFE, row: 1869
stock:PFE, row: 1870
stock:PFE, row: 1871
stock:PFE, row: 1872
stock:PFE, row: 1873
stock:PFE, row: 1874
stock:PFE, row: 1875
stock:PFE, row: 1876
stock:PFE, row: 1877
stock:PFE, row: 1878
stock:PFE, row: 1879
stock:PFE, row: 1880
stock:PFE, row: 1881
stock:PFE, row: 1882
stock:PFE, row: 1883
stock:PFE, row: 1884
stock:PFE, row: 1885
stock:PFE, row: 1886
stock:PFE, row: 1887
stock:PFE, row: 1888
stock:PFE, row: 1889
stock:PFE, row: 1890
stock:PFE, row: 1891
stock:PFE, row: 1892
stock:PFE, row: 1893
stock:PFE, row: 1894
stock:PFE, row: 1895
stock:PFE, row: 1896
stock:PFE, row: 1897
stock:PFE, row: 1898
stock:PFE, row: 1899
stock:PFE, row: 1900
stock:PFE, row: 1901
stock:PFE, row: 1902
stock:PFE, row: 1903
stock:PFE, row: 1904
stock:PFE, row: 1905
stock:PFE, row: 1906
stock:PFE, row: 1907
stock:PFE, row: 1908
stock:PFE, row: 1909
stock:PFE, row: 1910
stock:PFE, row: 1911
stock:PFE, row: 1912
stock:PFE, ro

stock:PFE, row: 2823
stock:PFE, row: 2824
stock:PFE, row: 2825
stock:PFE, row: 2826
stock:PFE, row: 2827
stock:PFE, row: 2828
stock:PFE, row: 2829
stock:PFE, row: 2830
stock:PFE, row: 2831
stock:PFE, row: 2832
stock:PFE, row: 2833
stock:PFE, row: 2834
stock:PFE, row: 2835
stock:PFE, row: 2836
stock:PFE, row: 2837
stock:PFE, row: 2838
stock:PFE, row: 2839
stock:PFE, row: 2840
stock:PFE, row: 2841
stock:PFE, row: 2842
stock:PFE, row: 2843
stock:PFE, row: 2844
stock:PFE, row: 2845
stock:PFE, row: 2846
stock:PFE, row: 2847
stock:PFE, row: 2848
stock:PFE, row: 2849
stock:PFE, row: 2850
stock:PFE, row: 2851
stock:PFE, row: 2852
stock:PFE, row: 2853
stock:PFE, row: 2854
stock:PFE, row: 2855
stock:PFE, row: 2856
stock:PFE, row: 2857
stock:PFE, row: 2858
stock:PFE, row: 2859
stock:PFE, row: 2860
stock:PFE, row: 2861
stock:PFE, row: 2862
stock:PFE, row: 2863
stock:PFE, row: 2864
stock:PFE, row: 2865
stock:PFE, row: 2866
stock:PFE, row: 2867
stock:PFE, row: 2868
stock:PFE, row: 2869
stock:PFE, ro

stock:WMB, row: 1068
stock:WMB, row: 1069
stock:WMB, row: 1070
stock:WMB, row: 1071
stock:WMB, row: 1072
stock:WMB, row: 1073
stock:WMB, row: 1074
stock:WMB, row: 1075
stock:WMB, row: 1076
stock:WMB, row: 1077
stock:WMB, row: 1078
stock:WMB, row: 1079
stock:WMB, row: 1080
stock:WMB, row: 1081
stock:WMB, row: 1082
stock:WMB, row: 1083
stock:WMB, row: 1084
stock:WMB, row: 1085
stock:WMB, row: 1086
stock:WMB, row: 1087
stock:WMB, row: 1088
stock:WMB, row: 1089
stock:WMB, row: 1090
stock:WMB, row: 1091
stock:WMB, row: 1092
stock:WMB, row: 1093
stock:WMB, row: 1094
stock:WMB, row: 1095
stock:WMB, row: 1096
stock:WMB, row: 1097
stock:WMB, row: 1098
stock:WMB, row: 1099
stock:WMB, row: 1100
stock:WMB, row: 1101
stock:WMB, row: 1102
stock:WMB, row: 1103
stock:WMB, row: 1104
stock:WMB, row: 1105
stock:WMB, row: 1106
stock:WMB, row: 1107
stock:WMB, row: 1108
stock:WMB, row: 1109
stock:WMB, row: 1110
stock:WMB, row: 1111
stock:WMB, row: 1112
stock:WMB, row: 1113
stock:WMB, row: 1114
stock:WMB, ro

stock:WMB, row: 2151
stock:WMB, row: 2152
stock:WMB, row: 2153
stock:WMB, row: 2154
stock:WMB, row: 2155
stock:WMB, row: 2156
stock:WMB, row: 2157
stock:WMB, row: 2158
stock:WMB, row: 2159
stock:WMB, row: 2160
stock:WMB, row: 2161
stock:WMB, row: 2162
stock:WMB, row: 2163
stock:WMB, row: 2164
stock:WMB, row: 2165
stock:WMB, row: 2166
stock:WMB, row: 2167
stock:WMB, row: 2168
stock:WMB, row: 2169
stock:WMB, row: 2170
stock:WMB, row: 2171
stock:WMB, row: 2172
stock:WMB, row: 2173
stock:WMB, row: 2174
stock:WMB, row: 2175
stock:WMB, row: 2176
stock:WMB, row: 2177
stock:WMB, row: 2178
stock:WMB, row: 2179
stock:WMB, row: 2180
stock:WMB, row: 2181
stock:WMB, row: 2182
stock:WMB, row: 2183
stock:WMB, row: 2184
stock:WMB, row: 2185
stock:WMB, row: 2186
stock:WMB, row: 2187
stock:WMB, row: 2188
stock:WMB, row: 2189
stock:WMB, row: 2190
stock:WMB, row: 2191
stock:WMB, row: 2192
stock:WMB, row: 2193
stock:WMB, row: 2194
stock:WMB, row: 2195
stock:WMB, row: 2196
stock:WMB, row: 2197
stock:WMB, ro

stock:FITB, row: 113
stock:FITB, row: 114
stock:FITB, row: 115
stock:FITB, row: 116
stock:FITB, row: 117
stock:FITB, row: 118
stock:FITB, row: 119
stock:FITB, row: 120
stock:FITB, row: 121
stock:FITB, row: 122
stock:FITB, row: 123
stock:FITB, row: 124
stock:FITB, row: 125
stock:FITB, row: 126
stock:FITB, row: 127
stock:FITB, row: 128
stock:FITB, row: 129
stock:FITB, row: 130
stock:FITB, row: 131
stock:FITB, row: 132
stock:FITB, row: 133
stock:FITB, row: 134
stock:FITB, row: 135
stock:FITB, row: 136
stock:FITB, row: 137
stock:FITB, row: 138
stock:FITB, row: 139
stock:FITB, row: 140
stock:FITB, row: 141
stock:FITB, row: 142
stock:FITB, row: 143
stock:FITB, row: 144
stock:FITB, row: 145
stock:FITB, row: 146
stock:FITB, row: 147
stock:FITB, row: 148
stock:FITB, row: 149
stock:FITB, row: 150
stock:FITB, row: 151
stock:FITB, row: 152
stock:FITB, row: 153
stock:FITB, row: 154
stock:FITB, row: 155
stock:FITB, row: 156
stock:FITB, row: 157
stock:FITB, row: 158
stock:FITB, row: 159
stock:FITB, r

stock:FITB, row: 990
stock:FITB, row: 991
stock:FITB, row: 992
stock:FITB, row: 993
stock:FITB, row: 994
stock:FITB, row: 995
stock:FITB, row: 996
stock:FITB, row: 997
stock:FITB, row: 998
stock:FITB, row: 999
stock:FITB, row: 1000
stock:FITB, row: 1001
stock:FITB, row: 1002
stock:FITB, row: 1003
stock:FITB, row: 1004
stock:FITB, row: 1005
stock:FITB, row: 1006
stock:FITB, row: 1007
stock:FITB, row: 1008
stock:FITB, row: 1009
stock:FITB, row: 1010
stock:FITB, row: 1011
stock:FITB, row: 1012
stock:FITB, row: 1013
stock:FITB, row: 1014
stock:FITB, row: 1015
stock:FITB, row: 1016
stock:FITB, row: 1017
stock:FITB, row: 1018
stock:FITB, row: 1019
stock:FITB, row: 1020
stock:FITB, row: 1021
stock:FITB, row: 1022
stock:FITB, row: 1023
stock:FITB, row: 1024
stock:FITB, row: 1025
stock:FITB, row: 1026
stock:FITB, row: 1027
stock:FITB, row: 1028
stock:FITB, row: 1029
stock:FITB, row: 1030
stock:FITB, row: 1031
stock:FITB, row: 1032
stock:FITB, row: 1033
stock:FITB, row: 1034
stock:FITB, row: 103

stock:FITB, row: 1986
stock:FITB, row: 1987
stock:FITB, row: 1988
stock:FITB, row: 1989
stock:FITB, row: 1990
stock:FITB, row: 1991
stock:FITB, row: 1992
stock:FITB, row: 1993
stock:FITB, row: 1994
stock:FITB, row: 1995
stock:FITB, row: 1996
stock:FITB, row: 1997
stock:FITB, row: 1998
stock:FITB, row: 1999
stock:FITB, row: 2000
stock:FITB, row: 2001
stock:FITB, row: 2002
stock:FITB, row: 2003
stock:FITB, row: 2004
stock:FITB, row: 2005
stock:FITB, row: 2006
stock:FITB, row: 2007
stock:FITB, row: 2008
stock:FITB, row: 2009
stock:FITB, row: 2010
stock:FITB, row: 2011
stock:FITB, row: 2012
stock:FITB, row: 2013
stock:FITB, row: 2014
stock:FITB, row: 2015
stock:FITB, row: 2016
stock:FITB, row: 2017
stock:FITB, row: 2018
stock:FITB, row: 2019
stock:FITB, row: 2020
stock:FITB, row: 2021
stock:FITB, row: 2022
stock:FITB, row: 2023
stock:FITB, row: 2024
stock:FITB, row: 2025
stock:FITB, row: 2026
stock:FITB, row: 2027
stock:FITB, row: 2028
stock:FITB, row: 2029
stock:FITB, row: 2030
stock:FITB

stock:CHRW, row: 206
stock:CHRW, row: 207
stock:CHRW, row: 208
stock:CHRW, row: 209
stock:CHRW, row: 210
stock:CHRW, row: 211
stock:CHRW, row: 212
stock:CHRW, row: 213
stock:CHRW, row: 214
stock:CHRW, row: 215
stock:CHRW, row: 216
stock:CHRW, row: 217
stock:CHRW, row: 218
stock:CHRW, row: 219
stock:CHRW, row: 220
stock:CHRW, row: 221
stock:CHRW, row: 222
stock:CHRW, row: 223
stock:CHRW, row: 224
stock:CHRW, row: 225
stock:CHRW, row: 226
stock:CHRW, row: 227
stock:CHRW, row: 228
stock:CHRW, row: 229
stock:CHRW, row: 230
stock:CHRW, row: 231
stock:CHRW, row: 232
stock:CHRW, row: 233
stock:CHRW, row: 234
stock:CHRW, row: 235
stock:CHRW, row: 236
stock:CHRW, row: 237
stock:CHRW, row: 238
stock:CHRW, row: 239
stock:CHRW, row: 240
stock:CHRW, row: 241
stock:CHRW, row: 242
stock:CHRW, row: 243
stock:CHRW, row: 244
stock:CHRW, row: 245
stock:CHRW, row: 246
stock:CHRW, row: 247
stock:CHRW, row: 248
stock:CHRW, row: 249
stock:CHRW, row: 250
stock:CHRW, row: 251
stock:CHRW, row: 252
stock:CHRW, r

stock:CHRW, row: 1293
stock:CHRW, row: 1294
stock:CHRW, row: 1295
stock:CHRW, row: 1296
stock:CHRW, row: 1297
stock:CHRW, row: 1298
stock:CHRW, row: 1299
stock:CHRW, row: 1300
stock:CHRW, row: 1301
stock:CHRW, row: 1302
stock:CHRW, row: 1303
stock:CHRW, row: 1304
stock:CHRW, row: 1305
stock:CHRW, row: 1306
stock:CHRW, row: 1307
stock:CHRW, row: 1308
stock:CHRW, row: 1309
stock:CHRW, row: 1310
stock:CHRW, row: 1311
stock:CHRW, row: 1312
stock:CHRW, row: 1313
stock:CHRW, row: 1314
stock:CHRW, row: 1315
stock:CHRW, row: 1316
stock:CHRW, row: 1317
stock:CHRW, row: 1318
stock:CHRW, row: 1319
stock:CHRW, row: 1320
stock:CHRW, row: 1321
stock:CHRW, row: 1322
stock:CHRW, row: 1323
stock:CHRW, row: 1324
stock:CHRW, row: 1325
stock:CHRW, row: 1326
stock:CHRW, row: 1327
stock:CHRW, row: 1328
stock:CHRW, row: 1329
stock:CHRW, row: 1330
stock:CHRW, row: 1331
stock:CHRW, row: 1332
stock:CHRW, row: 1333
stock:CHRW, row: 1334
stock:CHRW, row: 1335
stock:CHRW, row: 1336
stock:CHRW, row: 1337
stock:CHRW

stock:CHRW, row: 2141
stock:CHRW, row: 2142
stock:CHRW, row: 2143
stock:CHRW, row: 2144
stock:CHRW, row: 2145
stock:CHRW, row: 2146
stock:CHRW, row: 2147
stock:CHRW, row: 2148
stock:CHRW, row: 2149
stock:CHRW, row: 2150
stock:CHRW, row: 2151
stock:CHRW, row: 2152
stock:CHRW, row: 2153
stock:CHRW, row: 2154
stock:CHRW, row: 2155
stock:CHRW, row: 2156
stock:CHRW, row: 2157
stock:CHRW, row: 2158
stock:CHRW, row: 2159
stock:CHRW, row: 2160
stock:CHRW, row: 2161
stock:CHRW, row: 2162
stock:CHRW, row: 2163
stock:CHRW, row: 2164
stock:CHRW, row: 2165
stock:CHRW, row: 2166
stock:CHRW, row: 2167
stock:CHRW, row: 2168
stock:CHRW, row: 2169
stock:CHRW, row: 2170
stock:CHRW, row: 2171
stock:CHRW, row: 2172
stock:CHRW, row: 2173
stock:CHRW, row: 2174
stock:CHRW, row: 2175
stock:CHRW, row: 2176
stock:CHRW, row: 2177
stock:CHRW, row: 2178
stock:CHRW, row: 2179
stock:CHRW, row: 2180
stock:CHRW, row: 2181
stock:CHRW, row: 2182
stock:CHRW, row: 2183
stock:CHRW, row: 2184
stock:CHRW, row: 2185
stock:CHRW

TER       ADM      TMUS      IDXX       LNC       WST  \
Date                                                                     
2011-01-04  0.001348  0.000166  0.001597  0.000327  0.000036  0.000617   
2011-01-05  0.001227  0.000164  0.001757  0.000250 -0.000017  0.000121   
2011-01-06  0.001075  0.000164  0.001858  0.000256 -0.000047  0.000425   
2011-01-07  0.001143  0.000173  0.001728  0.000243 -0.000105  0.000432   
2011-01-10  0.001210  0.000177  0.001574  0.000107 -0.000148  0.000283   

                AMZN       DOV      ORLY       BSX       USB       ETR  \
Date                                                                     
2011-01-04  0.000139  0.000365  0.000581 -0.000579  0.000009 -0.000400   
2011-01-05  0.000148  0.000324  0.000462 -0.000610 -0.000015 -0.000384   
2011-01-06  0.000156  0.000312  0.000465 -0.000605 -0.000038 -0.000371   
2011-01-07  0.000164  0.000271  0.000400 -0.000660 -0.000070 -0.000388   
2011-01-10  0.000178  0.000228  0.000405 -0.000677 -0.000097 -0.000364   

                 AZO       PVH      BIIB       ETN       PFE       WMB  \
Date                                                                     
2011-01-04  0.000603  0.000731  0.000076  0.000537 -0.000139  0.000029   
2011-01-05  0.000531  0.000608  0.000054  0.000513 -0.000158 -0.000022   
2011-01-06  0.000521  0.000593  0.000067  0.000533 -0.000141 -0.000039   
2011-01-07  0.000494  0.000450  0.000080  0.000516 -0.000135 -0.000093   
2011-01-10  0.000480  0.000319  0.000094  0.000495 -0.000125 -0.000056   

                FITB      CHRW  
Date                            
2011-01-04  0.000440  0.000300  
2011-01-05  0.000376  0.000258  
2011-01-06  0.000364  0.000279  
2011-01-07  0.000311  0.000294  
2011-01-10  0.000226  0.000308

### 4.3 Estimating STR Betas

We estimate stocks's sensitivity to the STR factor on daily data using linear regression (implemented in the str_betas function from computation_functions.py script).

In [16]:
str_betas = cf.str_betas(excess_returns, reversal)
str_betas.head()

Elapsed time: 11.516 seconds.


TER       ADM      TMUS      IDXX       LNC       WST  \
Date                                                                     
2011-01-04  0.006474  0.003697  0.002748  0.006012  0.008347  0.004482   
2011-01-05  0.006487  0.003700  0.002796  0.006016  0.008376  0.004481   
2011-01-06  0.006628  0.003717  0.002669  0.006016  0.008514  0.004362   
2011-01-07  0.006559  0.003533  0.002440  0.006058  0.008726  0.004329   
2011-01-10  0.006578  0.003576  0.002356  0.005957  0.008675  0.004310   

                AMZN       DOV      ORLY       BSX       USB       ETR  \
Date                                                                     
2011-01-04  0.001955  0.005945  0.003569  0.003264  0.008570  0.003547   
2011-01-05  0.001953  0.005955  0.003561  0.003260  0.008579  0.003555   
2011-01-06  0.001920  0.005987  0.003560  0.003248  0.008767  0.003486   
2011-01-07  0.001818  0.006149  0.003682  0.003436  0.008967  0.003566   
2011-01-10  0.001852  0.006093  0.003688  0.003411  0.008905  0.003603   

                 AZO       PVH      BIIB       ETN       PFE       WMB  \
Date                                                                     
2011-01-04  0.002324  0.005968  0.003877  0.008848  0.002873  0.005798   
2011-01-05  0.002312  0.005966  0.003875  0.008854  0.002904  0.005811   
2011-01-06  0.002327  0.006145  0.003816  0.008776  0.002756  0.005903   
2011-01-07  0.002362  0.006464  0.003794  0.008825  0.002720  0.006183   
2011-01-10  0.002343  0.006376  0.003816  0.008792  0.002749  0.006236   

                FITB      CHRW  
Date                            
2011-01-04  0.012959  0.003223  
2011-01-05  0.012982  0.003226  
2011-01-06  0.013103  0.003114  
2011-01-07  0.013290  0.003030  
2011-01-10  0.013168  0.003049

### 4.4 Transforming Data to Monthly Values

In [17]:
# firstly restricting to same time range, since to compute betas, we needed to give up of some data
size = size.iloc[253:, :]
excess_returns = excess_returns.iloc[253:, :]
market = market.iloc[253:, :]
reversal = reversal.iloc[253:, :]

In [18]:
# to monthly
monthly_size = sp.daily_to_monthly(size)
monthly_returns = sp.daily_to_monthly(excess_returns)
monthly_market = sp.daily_to_monthly(market)
monthly_reversal = sp.daily_to_monthly(reversal)
monthly_mkt_betas = sp.daily_to_monthly(mkt_betas)
monthly_size_betas = sp.daily_to_monthly(size_betas)
monthly_str_betas = sp.daily_to_monthly(str_betas)

## 5 Fama-MacBeth
### 5.1 Estimating Gamma Coefficients
Now we perform the second part of the regression - estimating the coefficients for excess returns on beta factors we estimated above in section 4.

In [19]:
gammas = cf.fama_macbeth(monthly_returns, monthly_mkt_betas, monthly_size_betas, monthly_str_betas)
gammas.head()

intercept       mkt      size       str
Date                                               
2011-01-31  -0.003352  0.092847  1.919403  0.333450
2011-02-28  -0.000650  0.265117  0.942599 -0.249414
2011-03-31   0.002725 -0.030486  1.386462 -0.368425
2011-04-30   0.005715 -0.060669 -1.655870 -0.463949
2011-05-31   0.000667 -0.178482  1.676451 -0.001218

We stored all the coefficients into the dataframe above (gammas), and finally do the statistics on obtained coefficients. The results of Fama-MacBeth regression are displayed below.

In [20]:
results = cf.results_table(gammas)
results

intercept       mkt      size       str
statistics                                        
mean       -0.001332 -0.023097  0.154813  -0.00407
st. dev     0.004145  0.368388  1.547875  0.351767
t-stat     -0.321387 -0.062698  0.100017 -0.011569

We estimated coefficients and t-stats for the case when we are controlling all sort variables (market, size, and STR). For the size - we added epsilon=0.000001 to the market cap and then took logarithm of it, to prevent the ill behaviour of the estimates, since the log(0)= - infinity.

From the Fama-Macbeth Regression, the values for intercept, market beta, size, and short-term reversal are -0.001332, -0.023097, 0.154813 and -0.00407, respectively. The regression model for excess returns is therefore constructed as follows: 

$$E(R) = -0.023097*mkt~beta + 0.154813* size -0.00407*str - 0.001332$$

From the results, it could appear that creating the portfolio with only big-size companies stocks could "beat the market", however, looking at the t-stat, we can conclude that none of the coefficients are significant, which is consistent with the theory.